### BIOL4289
# Phylogenetics and SARS-CoV-2

<br/>

#### Art Poon
### Western University

<small>
Departments of Pathology & Laboratory Medicine; Computer Science; Applied Mathematics; Microbiology and Immunology
</small>

<small><small>
<br/>
Slides published at http://slides.filogeneti.ca/html/BIOL4289.html
</small></small>

---

# What's the role of phylogenetics?

<table>
  <tr>
    <td style="font-size: 20pt;">
      <ul>
      <li>First complete genome sequence (~30K nt) <a href="https://virological.org/t/novel-2019-coronavirus-genome/319">released</a> January 10<sup>th</sup>.</li>
      <li>Presently over 800,000 genome records in GISAID.</li>
      <li>Exponential trajectory in number of genomes over time.</li>
      <li>Shorter delays between sample collection and data release.</li>
      </ul>
    </td>
    <td width="45%">
      <img src="/img/gisaid-dates.png"/>
    </td>
  </tr>
</table>

---

### 2020-01-10: Placement of SARS-CoV-2 in Sarbecoviruses

<img src="/img/sarscov2-origin.jpg" height="500px"/>

<small><small>
Image credit: R Lu <i>et al.</i> (Feb 2020).  Genomic characterisation and epidemiology of 2019 novel coronavirus: implications for virus origins and receptor binding.  Lancet 395: 565-574.
</small></small>

---

### 2020-01-23: Evidence of human-to-human transmission (Nextstrain)

<iframe src="https://nextstrain.org/narratives/ncov/sit-rep/2020-01-23?n=5" width="1000px" height="500px">
</iframe>

<small><small>
Image credit: T Bedford <i>et al.</i> (2020) Genomic analysis of nCoV spread. Situation report 2020-01-23.
</small></small>

---

### 2020-03-06: Root-to-tip estimation of clock and root date
<iframe src="https://rambaut.github.io/ncov-2019/ncov_ml_tree_rtt.html" width="1000px" height="500px">
</iframe>

<small><small>
Image credit: Andrew Rambaut (2020) <a href="https://virological.org/t/phylodynamic-analysis-176-genomes-6-mar-2020/356">Phylogenetic analysis of nCoV-2019 genomes</a>.  https://virological.org
</small></small>

---

### 2020-03-13: Phylodynamic estimates of R<sub>0</sub>
![](https://virological.org/uploads/default/original/1X/8f21f08be4ef14d94441e26c058332e72536ea53.png)

<small><small>
Image credit: TG Vaughan <i>et al.</i> (2020) <a href="https://virological.org/t/phylodynamic-analyses-of-outbreaks-in-china-italy-washington-state-usa-and-the-diamond-princess/439">Phylodynamic Analyses of outbreaks in China, Italy, Washington State (USA), and the Diamond Princess</a>.  https://virological.org
</small></small>

---

### 2020-03-15: Genomic surveillance of incoming virus (Nextstrain)

<iframe src="https://nextstrain.org/community/blab/ncov-cryptic-transmission/introductions" width="1000px" height="500px">
</iframe>

<small><small>
Image credit: T Bedford <i>et al.</i> (2020) Cryptic transmission of SARS-CoV-2 in Washington State.  Science 370: <a href="https://science.sciencemag.org/content/370/6516/571">571-575</a>.
</small></small>

---

### 2020-04-17: Dynamic nomenclature proposal of SARS-CoV-2 lineages

<table>
  <tr>
    <td style="font-size: 20pt;">
      <ul style="font-size: 18pt;">
      <li>Researchers use a maximum likelihood phylogeny to propose a nomenclature of SARS-CoV-2 genomes .</li>
      <li>A lineage must:</li>
      <ol>
        <li>have one or more synapomorphy.</li>
        <li>comprise at least 5 genomes (at least 95% coverage)</li>
        <li>have at least one internal synapomorphy</li>
        <li>bootstrap support of at least 70%.</li>
      </ol>
      </ul>
    </td>
    <td width="45%">
      <img src="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41564-020-0770-5/MediaObjects/41564_2020_770_Fig1_HTML.png" height="400px"/>
    </td>
  </tr>
</table>

<small><small>
Image credit: A Rambaut <i>et al.</i> (2020) A dynamic nomenclature proposal for SARS-CoV-2 lineages to assist genomic epidemiology.  Nat Microbiol 5: <a href="https://www.nature.com/articles/s41564-020-0770-5">1403-1407</a>.
</small></small>

---

### 2020-04-06:  Tracking patterns of selection

Using cutting-edge phylogenetic methods for detecting selection
<img src="https://virological.org/uploads/default/original/1X/a68acfdbee92ddcda954830610658a94af1188c0.png"/>

<small><small>
Image credit: Sergei Pond (2020) Selection analysis of GISAID SARS-CoV-2 data.  Nat Microbiol 5: <a href="https://www.nature.com/articles/s41564-020-0770-5">1403-1407</a>.
</small></small>

---

### 2020-05-27:  Fixation of D614G becomes apparent


<img src="https://virological.org/uploads/default/original/2X/b/b40d5137063807921db2a707c4acbdd97b0f66a2.jpeg" height="400px"/>

<small><small>
Image credit: Nextstrain screen capture <a href="https://virological.org/t/spike-protein-d614g-variant-paper-published-in-cell-today/526">posted</a> by Brain Foley, virological.org (2020).
</small></small>

---

### 2020-12-16:  First variant of concern (B.1.1.7)

<img src="https://virological.org/uploads/default/original/2X/8/80b48244e0484fe8ba8b30722792f4b45462e314.png" height="400px"/>

<small><small>
Image credit: A Rambaut <i>et al.</i> (2020) <a href="https://virological.org/t/preliminary-genomic-characterisation-of-an-emergent-sars-cov-2-lineage-in-the-uk-defined-by-a-novel-set-of-spike-mutations/563">Preliminary genomic characterisation of an emergent SARS-CoV-2 lineage in the UK defined by a novel set of spike mutations</a>.  https://virological.org
</small></small>

---

### 2021-02-02:  WHO convenes working group to resolve nomenclatures

"B.1.1.7" aka "20I/501Y.V1" aka "VOC 202012/01" aka "UK variant"
![](https://pbs.twimg.com/media/Er_AAj6XAAADQ1E?format=jpg&name=4096x4096)

---


# Challenges for phylogenetics

* Enormous amounts of data
  * Tree space increases super-exponentially with sample size.
* Short epidemiological time scale &hairsp;&#8658;&hairsp; most infections differ by zero to one substitutions from source.
  * Extensive polytomies (unresolved internal nodes in tree)
  * Exacerbated by missing data.

---

<b>CoVizu</b> is an open-source project to provide a public interface to visualize the global diversity of SARS-CoV-2 genomes.
<img src="/img/covizu-screen.png" height="550px"/>

---

# Database

* Under a GISAID Data Access Agreement, CoVizu is provisioned by a custom data feed.
  <img src="/img/covizu-gisaid.png" height="450px"/>

---

# Sequence alignment

* Multiple alignment is prohibitively time-consuming.
* Pairwise alignment to a reference and discarding insertions is *de facto* standard.
* We use a Python wrapper with *minimap2* to align roughly 1,600 genomes/second.

<img src="/img/minimap2.png" height="167px"/>

---

# Genome encoding

* Instead of working with aligned sequences, we extract all differences from a reference genome from *minimap2* output stream as *feature subsets*.
  * Highly compact representation of genome (~10 features versus 30,000 nt).
  * Retain all insertion and deletions in addition to substitutions.
* Similar to mutation annotation of tree strategy by [Yatish Turakhia](https://github.com/yatisht/usher) (UShER, UCSC).

---

# Data cleaning

<table>
<tr>
<td>
<ul>
<li>We set a low tolerance for uncalled bases (<300 <tt>N</tt>s).</li>
<li>Exclude problematic sites as defined by <a href="https://github.com/W-L/ProblematicSites_SARS-CoV2">VCF file</a> maintained by <a href="https://www.ebi.ac.uk/research/goldman">Nick Goldman's group @EBI</a>.</li>
<li>Exclude sequences with excessive divergence given collection date (right), assuming molecular clock of <tt>0.0655</tt> subs/yr and <tt>2019-12-01</tt> T<sub>MRCA</sub>.</li>
</ul>
</td>
<td style="width: 45%">
<img src="https://user-images.githubusercontent.com/1109328/95021105-0a450800-063d-11eb-9e90-0a4e0f76c382.png"</li>
<small><small>
Upper and lower 95% quantiles (dashed, Poisson) around clock expectation (solid) and observed numbers of genetic differences (points) in SARS-CoV-2 genomes.
</small></small>
</td>
</tr>
</table>

---

<table>
  <tr>
    <td style="vertical-align: middle;">
      <h1>Build time-scaled tree</h1>
      <ul>
        <li>Select representative genomes for all PANGO lineages (n=878; past filters, most recent sample).</li>
        <li>Run <a href="http://www.microbesonline.org/fasttree/">FastTree2</a> on MSA and rescaled with <a href="https://github.com/neherlab/treetime">TreeTime</a>.</li>
      </ul>
      <img src="/img/sc2-rtt.svg" height="250px"/>
    </td>
    <td width="40%">
      <img src="/img/timetree.svg" height="650px"/>
    </td>
  </tr>
</table>


---

# Genetic distances (within lineage)

* Assume all differences are equally likely to occur
  * Ignore multiple hits!
  * Not enough data to quantify rate variation among sites.
* Compute [symmetric difference](https://en.wikipedia.org/wiki/Symmetric_difference) between feature subsets.
  * *e.g.*, {1,4,128} &Delta; {4,37,89} = {1,37,89,128}
  * Use integer indices for features.
  * Merge genomes with identical feature subsets.
  * Write output to temporary file to reduce RAM usage.

---

# Bootstrapping

* Sample the feature set union at random with replacement, 100 times.
* Apply the resulting feature weights to symmetric difference matrix.
  * Yields integer-valued pairwise distance matrix.
* Write to file as input for neighbor-joining tree reconstruction with [RapidNJ](https://birc.au.dk/software/rapidnj/).
* Process with MPI with own HPC cluster.

---

# Generate consensus trees

* 100 bootstrap NJ trees for every lineage
* BioPython's `majority_consensus` function is slow and wrong!
  * Had to code my own function (33x faster).

<table>
  <tr>
    <td>
      BioPython
      <img src="https://user-images.githubusercontent.com/1109328/97740083-1a72ca80-1ab7-11eb-9832-1cb67abaa77a.png" width="250px"/>
    </td>
    <td>
      CoVizu
      <img src="https://user-images.githubusercontent.com/1109328/97740037-09c25480-1ab7-11eb-836e-526f8916181e.png" width="250px"/>
    </td>
    <td>
      ML
      <img src="https://user-images.githubusercontent.com/1109328/97748535-fc5f9700-1ac3-11eb-89ac-e52494142a3c.png" width="250px"/>
    </td>
  </tr>
</table>

---

# Convert consensus trees to beadplot data

* Collapse tips with mean branch length <0.5.
  * Use to label internal nodes.
* Collect collapsed polytomies into variants.
* Unlabeled internal nodes are unsampled variants.

<img src="/img/tree2beadplot.png" height="200px"/>

---


# Beadplots

<img src="/img/beadplot.png" height="300px8"/>

* Variants are represented by horizontal line segments
* Beads indicate samples from a particular date

---

# Front-end demonstration

* Implemented in JavaScript/D3js.
* <a href="https://filogeneti.ca/covizu" target="_blank">Live instance</a>
