### BIOL4289
# Phylogenetics and SARS-CoV-2

<br/>

#### Art Poon
### Western University

<small>
Departments of Pathology & Laboratory Medicine; Computer Science;  Microbiology and Immunology
</small>

<small><small>
<br/>
Slides published at http://slides.filogeneti.ca/html/BIOL4289.html
</small></small>

---

# What's the role of phylogenetics?

<table>
  <tr>
    <td style="font-size: 20pt;">
      <ul>
      <li>First complete genome sequence (~30K nt) <a href="https://virological.org/t/novel-2019-coronavirus-genome/319">released</a> January 10<sup>th</sup>.</li>
      <li>Presently over 15 million genome records in GISAID.</li>
      <li>Exponential trajectory in number of genomes over time.</li>
      <li>Shorter delays between sample collection and data release.</li>
      </ul>
    </td>
    <td width="45%">
      <img src="/img/gisaid.svg" width="500px"/>
    </td>
  </tr>
</table>

---

### 2020-01-10: Placement of SARS-CoV-2 in Sarbecoviruses

<img src="/img/sarscov2-origin.jpg" height="500px"/>

<small><small>
Image credit: R Lu <i>et al.</i> (Feb 2020).  Genomic characterisation and epidemiology of 2019 novel coronavirus: implications for virus origins and receptor binding.  Lancet 395: 565-574.
</small></small>

---

### 2020-01-23: Evidence of human-to-human transmission (Nextstrain)

<iframe src="https://nextstrain.org/narratives/ncov/sit-rep/2020-01-23?n=5" width="1000px" height="500px">
</iframe>

<small><small>
Image credit: T Bedford <i>et al.</i> (2020) Genomic analysis of nCoV spread. Situation report 2020-01-23.
</small></small>

---

### 2020-03-06: Root-to-tip estimation of clock and root date
<img src="/img/rambaut-virological.png"/>

<small><small>
Image credit: Andrew Rambaut (2020) <a href="https://virological.org/t/phylodynamic-analysis-176-genomes-6-mar-2020/356">Phylogenetic analysis of nCoV-2019 genomes</a>.  https://virological.org
</small></small>

---

### 2020-03-13: Phylodynamic estimates of R<sub>0</sub>
![](https://virological.org/uploads/default/original/1X/8f21f08be4ef14d94441e26c058332e72536ea53.png)

<small><small>
Image credit: TG Vaughan <i>et al.</i> (2020) <a href="https://virological.org/t/phylodynamic-analyses-of-outbreaks-in-china-italy-washington-state-usa-and-the-diamond-princess/439">Phylodynamic Analyses of outbreaks in China, Italy, Washington State (USA), and the Diamond Princess</a>.  https://virological.org
</small></small>

---

### 2020-03-15: Genomic surveillance of incoming virus (Nextstrain)

<iframe src="https://nextstrain.org/community/blab/ncov-cryptic-transmission/introductions" width="1000px" height="500px">
</iframe>

<small><small>
Image credit: T Bedford <i>et al.</i> (2020) Cryptic transmission of SARS-CoV-2 in Washington State.  Science 370: <a href="https://science.sciencemag.org/content/370/6516/571">571-575</a>.
</small></small>


---

### Testing hypotheses on virus introductions

<small>
Based on what we know about the rates of virus mutation, transmission, and sampling, it is unlikely that it
</small>
<img src="/img/370_564_f1.jpeg" height="400px">

<small><small>
Image credit: Worobey <i>et al.</i> (2020) The emergence of SARS-CoV-2 in Europe and North America. <a href="https://www.science.org/doi/full/10.1126/science.abc8169">Science 370: 564-570</a>.
</small></small>

---

### 2020-04-17: Dynamic nomenclature proposal of SARS-CoV-2 lineages

<table>
  <tr>
    <td style="font-size: 20pt;">
      <ul style="font-size: 18pt;">
      <li>Researchers use a maximum likelihood phylogeny to propose a nomenclature of SARS-CoV-2 genomes .</li>
      <li>A lineage must:</li>
      <ol>
        <li>have one or more synapomorphy.</li>
        <li>comprise at least 5 genomes (at least 95% coverage)</li>
        <li>have at least one internal synapomorphy</li>
        <li>bootstrap support of at least 70%.</li>
      </ol>
      </ul>
    </td>
    <td width="45%">
      <img src="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41564-020-0770-5/MediaObjects/41564_2020_770_Fig1_HTML.png" height="400px"/>
    </td>
  </tr>
</table>

<small><small>
Image credit: A Rambaut <i>et al.</i> (2020) A dynamic nomenclature proposal for SARS-CoV-2 lineages to assist genomic epidemiology.  Nat Microbiol 5: <a href="https://www.nature.com/articles/s41564-020-0770-5">1403-1407</a>.
</small></small>

---

### 2020-04-06:  Tracking patterns of selection

<small>
Using cutting-edge phylogenetic methods for detecting selection
</small>
<img src="https://virological.org/uploads/default/original/1X/a68acfdbee92ddcda954830610658a94af1188c0.png"/>

<small><small>
Image credit: Sergei Pond (2020) Selection analysis of GISAID SARS-CoV-2 data.  Nat Microbiol 5: <a href="https://www.nature.com/articles/s41564-020-0770-5">1403-1407</a>.
</small></small>

---

### 2020-04-29:  Initial release of Pangolin v1.0

![](/img/pangolin-github.png)

---

### 2020-05-27:  Fixation of D614G becomes apparent

<small>
First evidence of a selective sweep of a particular mutation. This spike protein mutation increases the efficiency of cell entry.
</small>

<img src="https://virological.org/uploads/default/original/2X/b/b40d5137063807921db2a707c4acbdd97b0f66a2.jpeg" height="400px"/>

<small><small>
Image credit: Nextstrain screen capture <a href="https://virological.org/t/spike-protein-d614g-variant-paper-published-in-cell-today/526">posted</a> by Brain Foley, virological.org (2020).
</small></small>

---

### 2020-08-15:  First new lineage proposal by open science

<small>
https://github.com/cov-lineages/pango-designation becomes a "crowd-sourced" system for tracking the emergence of new variants
</small>

![](/img/lineage-proposal.png)

---

### 2020-09-28:  Initial release of UShER

<img src="/img/usher-placement.png" height="500px">

<small><small>
Image credit: Turakhia <i>et al.</i> (2021) Ultrafast Sample placement on Existing tRees (UShER) enables real-time phylogenetics for the SARS-CoV-2 pandemic.  <a href="https://www.nature.com/articles/s41588-021-00862-7">Nature Genetics 53: 809-816</a>.
</small></small>

---

### 2020-12-16:  First variant of concern (B.1.1.7)

<img src="https://virological.org/uploads/default/original/2X/8/80b48244e0484fe8ba8b30722792f4b45462e314.png" height="400px"/>

<small><small>
Image credit: A Rambaut <i>et al.</i> (2020) <a href="https://virological.org/t/preliminary-genomic-characterisation-of-an-emergent-sars-cov-2-lineage-in-the-uk-defined-by-a-novel-set-of-spike-mutations/563">Preliminary genomic characterisation of an emergent SARS-CoV-2 lineage in the UK defined by a novel set of spike mutations</a>.  https://virological.org
</small></small>

---

### 2021-02-02:  WHO convenes working group to resolve nomenclatures

"B.1.1.7" aka "20I/501Y.V1" aka "VOC 202012/01" aka "UK variant"
![](https://pbs.twimg.com/media/Er_AAj6XAAADQ1E?format=jpg&name=4096x4096)

---

### 2021-03-17: First clear evidence of recombination

![](https://virological.org/uploads/default/19d4c46ebc4027541fda2a3c579ee9008a6e0238)

* UK surveillance program detects genomes carrying some mutations characteristic of B.1.1.7 (Alpha)
* Recombination is always occurring, but usually between similar genomes and difficult to detect.


<small><small>
Image source: https://virological.org/t/recombinant-sars-cov-2-genomes-involving-lineage-b-1-1-7-in-the-uk/658
</small></small>

---

### 2021-03-18: Timing the index case

<small>Investigators simulate trees to estimate the first infection to mid-Oct/Nov 2019.</small>
<img src="/img/372_412_f2.jpeg" height="500px">

<small><small>
Image source: Pekar <i>et al.</i> (2021) Timing the SARS-CoV-2 index case in Hubei province. <a href="https://www.science.org/doi/full/10.1126/science.abf8003">Science 372: 412-417</a>.
</small></small>

---

### 2021-08-05:  Elevated recombination rates in spike region

<small>
Using phylogenetic analysis (maximum parsimony), Turakhia <i>et al.</i> found extensive evidence of recombination in SARS-CoV-2 genomes.
</small>

<img src="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41586-022-05189-9/MediaObjects/41586_2022_5189_Fig2_HTML.png" height="450px">

<small><small>
Image source: Turakhia <i>et al.</i> (2022) Pandemic-scale phylogenomics reveals the SARS-CoV-2 recombination landscape.  <a href="https://www.nature.com/articles/s41586-022-05189-9">Nature 609: 994-997.</a>
</small></small>

---

### 2021-11-23:  First detection of Omicron by phylogenetic analysis

<table>
<tr>
<td>
  Original cov-lineages/pango-designation post identifying B.1.1.529.
  <img src="/img/peacock-omicron.png">
</td>
<td>
Tree displaying the initial Omicron genomes (red).
<img src="/img/FFJsU5qVcAQGzJl.jpeg">
</td>
</tr>
</table>

<small><small>
Image source: https://twitter.com/trvrb/status/1464353229941006336
</small></small>

---

### 2022-02-25: SARS-CoV-2 in white-tailed deer

<img src="https://www.biorxiv.org/content/biorxiv/early/2022/02/25/2022.02.22.481551/F2.large.jpg" height="400px"/>

<small><small>
Source: Pickering <i>et al.</i> (2022) Highly divergent white-tailed deer SARS-CoV-2 with potential deer-to-human transmission.  <a href="https://doi.org/10.1101/2022.02.22.481551">bioRxiv</a>.
</small></small>

---

### 2022-04-06: Zoonotic infection of farmed mink

<small>
Phylogenetic analysis of two outbreaks of SARS-CoV-2 in mink farms in the Netherlands and Denmark, suggesting a transient jump in the rate of evolution associated with switching hosts.
</small>
<img src="/img/vead002f2.jpeg" height="400px">

<small><small>
Image source:  Porter <i>et al.</i> (2023) Evolutionary rate of SARS-CoV-2 increases during zoonotic infection of farmed mink.  Virus Evol 9: <a href="https://academic.oup.com/ve/article/9/1/vead002/6980753">vead002</a>.
</small></small>

---

### 2022-07-26: Phylogenetic evidence of multiple zoonoses

<small>
Given what we know about the rates of virus mutation and transmission, the early appearance of lineages A and B is not consistent with a single origin.
</small>

<img src="/img/science.abp8337-f2.jpg" height="350px"/>

<small><small>
Image source: Pekar <i>et al.</i> (2022) The molecular epidemiology of multiple zoonotic origins of SARS-CoV-2.  <a href="https://www.science.org/doi/full/10.1126/science.abp8337">Science 377: 960-966.</a>
</small></small>

---

### 2022-12-04: Extensive convergent evolution in Omicron

<table>
  <tr>
    <td>
      <ul>
        <li>Circulating lineages may eventually become replaced by a recombinant (*e.g.*, XBB.1)</li>
        <li>Convergent evolution is also making it difficult to build accurate trees.</li>
      </ul>
      <small>
      Image source: Thomas Peacock, https://virological.org/t/sars-cov-2-evolution-post-omicron/911
      </small>
    </td>
    <td width="50%">
      <img src="https://virological.org/uploads/default/original/2X/a/a7355ebe1760868a8c93ed735f0503f45cda00ef.png"/>
    </td>
  </tr>
</table>

---

# Challenges for phylogenetics

* Enormous amounts of data
  * Tree space increases super-exponentially with sample size.
* Short epidemiological time scale &hairsp;&#8658;&hairsp; most infections differ by zero to one substitutions from source.
  * Extensive polytomies (unresolved internal nodes in tree)
  * Exacerbated by missing data.
* Extensive recombination makes it difficult to accurately represent the evolutionary history of SARS-CoV-2 with a tree.
  * A lot of recombination events between similar genomes are undetected.
