# Conference on the Evolutionary Genetics of Infectious Disease
## Unsupervised excursions into the deep evolutionary history of HIV-1 group M
### Art Poon
Western University

Department of Pathology and Laboratory Medicine, Department of Applied Mathematics, Department of Microbiology and Immunology

---


<iframe src="https://ourworldindata.org/grapher/number-of-people-living-with-hiv?time=1990..2016" style="width: 100%; height: 600px; border: 0px none;"></iframe>


---

Phylogeny of HIV/SIV *gag*
<img src="/img/HIV-tree.png" width="450px"/>

<small>
Excerpt of figure from Joy *et al* (2015) Global Virology I, Springer.
Generated by maximum likelihood.
</small>

---

# HIV-1 subtypes

<table>
<tr>
  <td>
    <ul>
      <li>Extraordinary diversity structured into subtypes.</li>
    </ul>
  <img src="/img/nomenclature-proposal.svg"/>
  </td>
  <td width="50%">
  <img src="/img/hiv1m-subtypes.svg"/>
  <br/>
  <small><small>
  (*above*) ML tree from LANL curated (2010) HIV-1/M env subtype references, aligned with MAFFT (excluding indel-rich regions) and reconstructed with RAxML.
  </small></small>

  </td>
</tr>
</table>

---

![](/img/tebit-arts.svg)

<small>
Figure from Tebit and Arts (2011) Lancet Inf Dis 11: 45-56.
</small>

---

# Impact of HIV-1 subtypes

* Understanding the global epidemiology of HIV-1/M
* Variation in rates of progression to AIDS (subtypes C, D)
<img src="/img/venner-arts.jpg" width="500px"/>
* Distinct mutational pathways to evolve drug resistance.

<small>
Figure from C Venner *et al.* (2016) EBioMedicine 13: 305-314.
</small>


---

# Subtyping

* Compare a sequence to a reference database of "pure" subtype sequences
* Identify recombinant as a combination of two or more parent reference sequences.
* Sliding window methods (BOOTSCAN, REGA)
* Phylogenetic methods (SCUEAL)
* Linear classifiers (COMET)

---

# Recombinant history of SIV

![](/img/bedford-siv.svg)

<small>
Figure from Bell and Bedford (2017) PLOS Pathog 13: e1006466.
</small>

---

# HIV-1 recombination

* Virus carries two copies of its genome.
* HIV reverse transcriptase switches templates during infection cycle.
* If a host cell is multiply infected, progeny virus can be recombinant.
* Effective recombination rate estimated at about 1%/genome/generation$^1$.
* Close to estimated mutation rate (0.2/genome/generation)$^2$.

<small>
$^1$ RA Neher *et al.* (2011) PLOS Comput Biol 6(1); $^2$ F Zanini *et al* (2017) Virus Evol 3(1).
</small>

---

# Circulating recombinant forms

<img src="/img/LANL-CRFs.svg"/>

---

# Dating the origin of HIV-1/M

<table>
<tr>
  <td>
    Root-to-tip regression
    <img src="/img/korber-fig1.png"/>
    <small>
    from B Korber *et al.* (2000) Science 288: 1789.
    </small>
  </td>
  <td width="50%">
    Bayesian sampling
    <img src="/img/worobey-fig.png"/>
    <small>
    from M Worobey *et al.* (2008) Nature 455: 661.
    </small>
  </td>
</tr>
</table>

---

# Molecular clock dating

<img src="/img/timetree.png" height="300px"/>
<img src="/img/timetree-scaled.png" height="250px"/>

---

# More genomes

* The number of near full-length HIV-1 genomes in public databases is nearing 10,000
* A recent global consortium (PANGEA-HIV) is generating an *additional* 10,000 HIV-1 genomes sampled throughout Africa.
* An emerging consensus is that recombination is more widespread than we thought.
> Given the inevitability of extensive recombination, it is highly unlikely that a single phylogeny can adequately represent the evolutionary history of HIV-1/M.

---

# Data collection

* Queried Genbank for HIV-1 sequences of minimum length 8,000nt ($n=7,816$)
* Manually reviewed all entries for *in vitro* clones, repeated samples, and non-group M variants.
* Reviewed associated literature for missing collection dates.
* Final total: $n=3,900$ genome sequences.

---


<table>
  <tr>
  <td style="vertical-align:middle;">
    <h1>Alignment</h1>
    <ul>
    <li>An alignment is a hypothesis about the evolutionary homology of specific residues between two sequences.</li>
    <li>The genomic diversity of HIV-1/M causes the alignment to explode with regions of low homology.</li>
    </ul>
  </td>
  <td width="40%">
    <img src="/img/mosaic/mafft_length.svg"/>
  </td>
  </tr>
  <tr>
    <td colspan=2><img src="/img/mosaic/randomized_3500_4.png"/></td>
  </tr>
</table>

---

# Procrustean alignment

<img src="/img/procrustes.png" width="400px"/>

* Aggressive pairwise alignment of each sequence against a reference genome.
* Any insertions relative to the reference are *discarded*.
* We need the most representative reference possible!

---

# Alignment-free clustering

* Use a p-spectrum kernel distance to cluster genomes:

$$k(s, t) = \sum_{u\in \mathcal{A}^6} C(u,s) C(u,t)$$

where $\mathcal{A}=\\{A,C,G,T\\}$ and $C(x,y)$ counts occurrences of substring $x$ in $y$.

* We use the standard cosine normalization:

$$k'(s,t) = \frac{k(s,t)}{\sqrt{k(s,s) k(t,t)}}$$

---

<section data-background="/img/mosaic/custom_dot.png" data-background-size="contain" />

---

# Consensus sequence

* Clustered genomes in this similarity graph with community detection algorithm (R *igraph* `cluster_leading_eigen`).
* Selected $n=32$ centres with highest degree centrality.
* Multiple alignment of central genomes to make majority-rule consensus for Procrustean alignment.
* Final alignment 10,171nt in length.
<img src="/img/mosaic/seq_accepted.png" height="200px"/>

---

# Sliding windows

* Extracted 81 windows of 500nt in steps of 100nt.
* For each window:
  * Reconstructed phylogeny by approximate ML ([FastTree2](http://www.microbesonline.org/fasttree/)).
  * Rooted phylogeny by root-to-tip regression ([rtt](https://cran.r-project.org/web/packages/ape/index.html))
  * Estimated time to the most recent common ancestor (TMRCA) under relaxed molecular clock ([LSD](https://github.com/tothuhien/lsd-0.3beta)).

---

TMRCA are variable and autocorrelated along the HIV-1 genome
![](/img/mosaic/tmrca.svg)

---

Similar results obtained using BEAST (data downsampled to ~300 tips).

![](/img/mosaic/beast.svg)

---

Demographic "skyline" reconstructions from BEAST
<img src="/img/mosaic/skylines.svg" height="600px"/>

---

# A deep recombination breakpoint?

* Why does HIV-1/M look older in *env*?
* Purifying selection may skew estimates of TMRCA.
  * No evidence of greater purifying selection in *gag* and *pol* versus *env*.
  * No evidence of saturation when comparing transitions to transversions.
* Examine signature of recombination.

---

# Recombination analysis

* Implemented a conventional distance-based recombination detection method in Python.
* Putative breakpoints less frequent within *env*.

<img src="/img/mosaic/RIP.svg" height="400px"/>

---

# What does this mean?

* The deep history of HIV-1/M is recombinant.
* Is the concept of "pure" subtypes still useful?
* HIV-1 subtypes and sub-subtypes are being re-assessed.
* Selective introgression of HIV-1 *env*: first contact with the virus envelope?

---

# A recombinant history

* How can we detect recombination if our reference points are themselves recombinant?

* Postulate 1: We cannot meaningfully describe recombinants as a mixture of "pure" subtypes on the time scale of HIV-1/M.
* Postulate 2: The recombinant history is too complex to reconstruct, *e.g.*, reticulate phylogenies.
* Postulate 3: There is enough residual homology to infer the evolutionary relationships of genome fragments. 

---

# Back to networks

* Let the homology of fragments be represented by a similarity graph.
* Graph clusters correspond to subtypes.
* As we move along the genome, fragments may shift from one cluster to another due to recombination.
* This process is analogous to a dynamic social network.

---

# Stochastic blockmodeling

* Generative models for network communities.
* Each node is assigned to one of $K$ communities (blocks) &mdash; block membership is a latent (unobservable) state of the node.
* Different edge probabilities within and between blocks.
* Transition of nodes among blocks over time is described by a Markov chain.

---

# Data reduction

* SBM becomes computationally non-feasible with the complete data set.
* We reduced the $n=3,900$ genome sequences to the most diverse $n=500$.
* Progressive "pruning" of the shortest branches from the ML phylogeny.

---

# Constructing a graph

* We calculated the Tamura-Nei (1993; TN93) genetic distance for every pair of fragments in a window.
* Used the lower 10% quantile of the TN93 distribution as the graph-defining threshold for each window.
* Processed the series of graphs along the HIV-1 genome with dynamic stochastic block models (R package [dynsvm](https://cran.r-project.org/web/packages/dynsbm/index.html); Matias and Miele).

---

# example of a graph

---

# Optimal K=25

* Integrated classification likelihood (ICL) criterion avoids overestimation of the number of clusters by penalizing likelihood$^1$.

<img src="/img/dynsbm/ICL.svg" height="400px"/>

<small>$^1$C Biernacki *et al.* (1998) Assessing a Mixture Model for Clustering with the Integrated Classification Likelihood. INRIA.</small>

---

<section data-background="/img/dynsbm/heatmap.svg" data-background-size="contain" />

---

<section data-background="/img/dynsbm/cake.png" data-background-size="contain" />

---

# Communities are fairly stable
![](/img/dynsbm/barplots.svg)

---


<section data-background="/img/dynsbm/subtypes.svg" data-background-size="contain">
<h1>Communities correlate with major subtypes</h1>
<br/><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/>
</section>

---

# How many genomes are recombinant?

---

**Community transitions are consistent with "known" recombinants.**

<img src="/img/dynsbm/breakpoints.svg" width="800px"/>
<small>
Plot excludes "blips" (transitions from A to B immediately followed by a revision from B to A in the next window).
</small>

---

# A new nomenclature proposal?

* Unsupervised clustering of genomic variation may define new centres of diversity.
* Can we select new reference points for characterizing recent recombinants?


---

![](/img/lab-photo.jpg)

<small>
Left to right: Dr. Mariano Avino, **Dr. Abayomi Olabode**, Garway Ng, Lisa-Monique Edward, John Palmer, Connor Chato, *Art and Fiona*, David Dick, Laura Mu&ntilde;oz-Baena.  (Photo credit: Marin Poon.)
<small>

---

<table>
  <tr>
    <td><img src="/img/OGI_Logo2015.png" height="150"/></td>
    <td><img src="/img/GenomeCanadaLogo.png" height="150"/></td>
    <td><img src="/img/NIH_NIAID.jpg" height="200"></td>
  </tr>
  <tr>
    <td><img src="/img/cihr.png"  height="200"/></td>
    <td><img src="/img/NSERC_RGB.png"/></td>
  </tr>
</table>

