# MIMM4750G
## Diversity and rates of evolution
![](https://imgs.xkcd.com/comics/evolving.png)

---

# Measuring diversity

* Genetic distances are useful
* What if there are many sequences (lots of pairwise comparisons!)
* Next-generation sequencing yields thousands to millions of reads.
* What if the sequences are very long?
* Bacterial genomes can be over 14Mbp long.
* Shortest animal genome is about 19.6Mbp (*Pratylenchus coffeae*, parasitic nematode of plants).

---

# Simple diversity measures

* Proportion of polymorphic sites in an alignment.
* Nucleotide diversity - the expected p-distance ($p$) between a random pair of sequences:
  `$$\pi = \sum_{i}\sum_{j} f_i f_j p_{ij}$$`
  where $f_i$ is the frequency of the $i$-th sequence variant.
* Shannon entropy - very common in bioinformatics (see next).

---

# Shannon Entropy

* Based on information theory, Shannon entropy is calculated from the frequencies of variants indexed by $i$:

  $$ S = -\sum_{i} p_i \log(p_i)$$

* If most frequencies are near zero, $S$ approaches 0.
* $S$ is greatest when frequencies are equal.
* Often averaged across nucleotide or amino acid sites of an alignment.

---

Application of Shannon entropy to characterize 16S rRNA gene diversity using PacBio (long read) and MiSeq (short read) NGS platforms.
<img src="/img/Clostridiales.svg" width="600px"/>

<small>
Figure from O Franz&eacute;n *et al.* (2015) Microbiome 3:43.
</small>

---

# Limitations of diversity measures

* Many diversity measures assume every sequence is an independent observation.
* We can badly overestimate diversity if most sequences inherited their variation from the same common ancestors.
* Consider two infections that differ by one nucleotide. 
  * Infection A is ancestral to 10 other infections. 
  * Infection B is ancestral to another 10.
  * If we compare this site across these 20 descendants, it will look very diverse!

---

<!--
# Using trees

* Building a tree relating sequences enables us to distinguish between *identity by state* and *identity by descent*.
* Identity by descent:  different infections inherited the variant from their common ancestor.
* Identity by state: variant arose independently in different infections.
* This is key information for reconstructing how many times a mutation occurred.

-->

# Rates of evolution

* Counting mutations is the key to measuring the rate of evolution
* **Why do rates matter?**
* Sites that evolve faster than others can reveal targets of selection.
* Rate of evolution may determine which variant survives.
* We can use the rate of evolution to extrapolate back in time.

---

Spread of *Escherichia coli* on a "megaplate" with gradients of antibiotic trimethoprim.
![](/img/megaplate5.gif)

<small>
Media from M Baym *et al* (2016) Science 353: 1147.
</small>

---

HIV-1 protease homodimer, colored by rate of evolution (blue fastest).
Protease inhibitor (yellow) bound at active site.
<img src="/img/foley-protease.jpeg" width="500px"/>

<small>
Image from Kuiken *et al.*, Los Alamos National Laboratory HIV Sequence Compendium 2003.
</small>

---

<section data-state="markov-slide">
    <h1>Modeling evolution</h1>
    <ul>
      <li>Recall that sequence evolution is often modeled as a *continuous-time Markov chain*</li>
      <li>Constant rate of evolution - exponential waiting times.</li>
    </ul>
    <center>
    <div id="markov" class="fig-container"
         data-fig-id="fig-markov"
         data-file="/include/markov-chain.html"
         style="height:300px">
    </div>
    <div></div>
    </center>
    <small>Based on JS by [Victor Powell](http://setosa.io/blog/2014/07/26/markov-chains/index.html)</small>
</section>

---

# Substitution models

* The Jukes-Cantor model can be expressed by the following rate matrix:

$$ \left( \begin{matrix}
  \ast & \mu & \mu & \mu \\\\
  \mu & \ast & \mu & \mu \\\\
  \mu & \mu & \ast & \mu \\\\
  \mu & \mu & \mu & \ast \\
  \end{matrix}\right)$$

* The diagonal entries $\ast$ are set to $-3\mu$ so that each row sums to 0.

<!--


# Rates into probabilities

* To convert this rate matrix into the probability that $X$ transitions into $Y$ after time $t$, we have to use *matrix exponentiation*.
* This is a computationally costly operation that accounts for all possible transitions that may occur in time $t$.
* For a rate matrix $Q$, the probability matrix is:

  $$P(t) = \exp^{Q t}$$
* This is where the confounding of rate and time appears.
-->

---

# Other models

* The Hasegawa-Kishino-Yano (HKY85) model allows for unequal base frequencies ($\pi_i$) and a transition/transversion rate bias ($\kappa$).

$$\begin{matrix}
 & \begin{matrix}A\;&\;C\;&\;G\;&\;T\end{matrix} \\\\
\begin{matrix}A\\\\C\\\\G\\\\T\end{matrix} & 
  \begin{pmatrix}
    \ast & \kappa \pi_C & \pi_G & \kappa\pi_T\\\\
    \kappa\pi_A & \ast & \kappa\pi_G & \pi_T\\\\
    \pi_A & \kappa\pi_C & \ast & \kappa\pi_T\\\\
    \kappa\pi_A & \pi_C & \kappa\pi_G & \ast
  \end{pmatrix}
\end{matrix}$$

---

# Generalized models

* In general, there are six rates for a time-reversible (symmetric rates) model:

  $$ \left( \begin{matrix}
  \ast & a & b & c \\\\
  a & \ast & d & e \\\\
  b & d & \ast & f \\\\
  c & e & f & \ast \\
  \end{matrix}\right)$$

  where these rates are assigned in alphabetical order &mdash; $a$ is the rate from `$A\leftrightarrow C$`, $b$ is `$A\leftrightarrow G$`, etc.

---

# Model specification

* PAUP* was a popular commercial software package for reconstructing phylogenies.
* It used a six-digit number ($abcdef$) to represent any kind of time-reversible nucleotide substitution model:
* *e.g.,* HKY85 becomes 010010.
* This scheme is still used by other software, such as HyPhy and PhyML.

---

# Why does the model matter?

* There are an enormous number of possible time-reversible models of nucleotide substitution.
* Using the wrong model (*model misspecification*) can bias estimates of other model parameters, *e.g.,* reconstructing the correct tree.
* The process of figuring out which model is best supported by the data is called *model selection*.

---

# Model selection

* We want to choose the model that has the best fit to the data.
* Adding parameters to the model improves the fit.
* We need to justify additional parameters!

  ![](/img/xkcd-curves-crop.png)

---

# Likelihood ratio test

* The *likelihood ratio test* (LRT) is a method of model selection that applies when one model is a special case of another.
* *e.g.,* the JC69 model is a special case of HKY85 where $\kappa=1$.
* If the likelihood of model 1 is $L_1$ and model 2 is $L_2$, then this test statistic:
  $$-2\log\left(\frac{L_1}{L_2}\right) = -2(\log L_1 - \log L_2)$$
  follows a $\chi^2_k$ distribution.
* $k$ is the difference in the number of parameters.

---

# Akaike information criterion

* What if the models are not nested?
* The Akaike information criterion (AIC) penalizes the model's likelihood by the number of parameters 

* *There is no statistical distribution*!  The best model minimizes the AIC.
  
  $$\text{AIC} = 2k - 2\log(L)$$


