# MIMM4750G
## Molecular clocks

![](https://imgs.xkcd.com/comics/estimating_time.png)

---

# What is a molecular clock?

<table>
  <tr>
  <td style="font-size: 18pt"><ul>
    <li>The molecular clock is a hypothesis that evolution at a molecular level (*e.g.,* proteins) occurs at a constant rate.</li>
    <li>Based on linear relationship between the time that species diverged from their common ancestor, and the number of amino acid differences in homologous proteins (*right*).</li>
    <li>The molecular clock does *not* "tick".</li>
  </ul></td>
  <td width="50%">
  ![](/img/Dickerson.png)
  <small>Figure from RE Dickerson (1971) J Mol Evol 1: 26.</small>
  </td>
  </tr>
</table>


---

# Neutral theory

<table>
  <tr>
  <td style="font-size: 18pt"><ul>
    <li>The [neutral theory](https://en.wikipedia.org/wiki/Neutral_theory_of_molecular_evolution): "the great majority of evolutionary changes at the molecular level [...] are caused not by Darwinian selection but by random drift"</li>
    <li>Without selection, nucleotide substitutions should be a random process at a constant rate.</li>
    <li>Today, this model is denoted the "strict" clock and is seldom supported by sufficient data.</li>
  </ul></td>
  <td width="35%">
  ![](https://www.genetics.org/content/genetics/202/4/1243/F1.medium.gif)
  <small>
  Motoo Kimura and James F. Crow
  <small>Image credit: Museum of Genetics, Federal University of Rio Grande do Sul, Brazil</small>
  </small>
  </td>
  </tr>
</table>

---

# Testing for a clock

<table>
  <tr>
  <td style="font-size: 18pt"><ul>
    <li>If a clock exists, then sequences should become more distant from the common ancestor as a *linear trend* over time.</li>
    <li>If the rate of evolution increases (decreases) over time, we expect this trend to be concave up (down).</li>
    <li>If the rate varies from one lineage to the next, the trend should be noisier.</li>
  </ul></td>
  <td width="55%">
  ![](http://currents.plos.org/outbreaks/files/2014/04/EBOV_cds_mb_path.png)
  <small>
  Root-to-tip regression of a Bayesian tree for concatenated coding regions of Ebola virus.
  <small>Image credit: Dudas and Rambaut (2014) PLOS Currents Outbreaks</small>
  </small>
  </td>
  </tr>
</table>



---

# Clocks and trees

* When we reconstruct a tree by fitting a model of evolution, the lengths of branches are measured in units of *expected numbers of substitutions*.

* This weird unit exists because most models are scaled to $\mu\times t$ (the rate of evolution multiplied by time).

* We say that the rate and time are *confounded* - it is impossible to estimate one parameter without knowing the true value of the other.

---

# Confound it!

<img src="/img/timetree.png" width="500px"/>

* A tree can be stretched back in time ($\uparrow t$) and explain the data with *exactly the same likelihood* if we decrease the rate ($\downarrow \mu$).
* Conversely, if we compress the tree forward in time ($\downarrow t$), we obtain the same likelihood by speeding up the rate ($\uparrow \mu$).

---

# Estimating the clock

* For infectious diseases, measurable evolution can occur on a time scale of weeks.
* This means that a branch of the tree can "grow" in the time between sampling different infections.
<img src="/img/timetree-scaled.png" width="400px"/>
* We can use sampling times to "pin down" the tree and prevent the free-scaling problem.

---

>Q1: Why doesn't this approach of "pinning down" tips of a molecular phylogeny in time work for animals like humans?

---

# Rooting the tree

* If we are directly measuring *time*, then we must impose a direction of time's flow on the tree.
* This means that we must locate the tree's *root* (the earliest point in time).
* Rooting trees is difficult!  By definition, it is the point that is the most distant in time &mdash; it cannot be directly observed.


---

# Outgroup rooting

* An *outgroup* is a species or infection that is not closely related to the sample set.
* We place the root at the point where the branch leading to the outgroup intersects the tree.
* If the outgroup is too *close*, then the root is too influenced by sampling bias (our choice of outgroup).
* If the outgroup is too *distant*, then where it intersects the tree is subject to random variation.

---

# Choosing an outgroup

* There is no objective framework to select an "optimal" outgroup.
* We often reach an informal consensus about which outgroup to use, without any other rationale.
  * *e.g.*, HIV-1 subtype D is an outgroup for HIV-1.
* If an early isolate of the pathogen is available, it is sometimes used as an outgroup.
  * *e.g.,* West Nile virus isolate B956 (isolated in 1940).
* **Use different outgroups to see how robust your conclusions are to choice of outgroup!**

---



<table>
  <tr>
    <td style="vertical-align:middle">
      <h1>Root-to-tip regression</h1>
      <ul>
        <li>We can simultaneously estimate the clock (rate of evolution) *and* locate the root.</li>
        <li>Try different placements of the root and look at the plot that results.</li>
        <li>The "best" root should give the cleanest (positive) linear trend.</li>
        <li>(*right*) Phylogeny and root-to-tip plot for Ebolavirus isolates from 2013-2016 epidemic in West Africa.</li>
      </ul>
    </td>
    <td width="35%">
      <img src="/img/nature19790.png" width="300px"/>
    </td>
  </tr>
</table>

---

# "Bad" root
<table>
  <tr>
    <td><img src="/img/ebola2-tree.png"/></td>
    <td><img src="/img/ebola2-rtt.png"/></td>
  </tr>
</table>

---

# "Good" root
<table>
  <tr>
    <td><img src="/img/ebola1-tree.png"/></td>
    <td><img src="/img/ebola1-rtt.png"/></td>
  </tr>
</table>

---

# Origin of HIV-1

* This concept was first used to estimate the origin date of HIV-1.

  <img src="/img/li-tanimura-sharp.png" width="600px"/>

<small>
Tree from Li, Tanimura and Sharp (1988) Mol Biol Evol 5: 313.
</small>

---

# Origin of HIV-1

<img src="/img/korber-fig1.png" width="400px"/>

<small>
Figure from B Korber *et al.* (2000) Science 288.
</small>

---

<img src="/img/inca-rtt.png" width="400px"/>
> INCA Q2.  In the "root to tip" plot above, write down which letters correspond to: (i) $t$; (2) $\mu t$; (3) $\mu$, (4) time of the root.

<small>
Figure taken from an analysis of respiratory syncytial virus by C Agoti *et al.* 2014, Emerg Infect Dis 20: 957.
</small>
