# MIMM4750G
## Metabarcoding and metagenomics

![](https://imgs.xkcd.com/comics/hofstadter.png)

---

# What's metabarcoding?

* So far we've talked about applications of NGS that focus on a single species (bacteria or virus).
* The massive throughput of NGS platforms allows us to target multiple species, or perform untargeted sequencing.
* A key challenge to untargeted sequencing is amplification.
* Amplification requires primers, but primers are usually specific to a particular sequence.

---

# Untargeted sequencing

* If we can't use template-specific primers, what *can* we do?
* Find a conserved region &mdash; shared across all, if not most &mdash; species, for amplification: but the region in between should be non-conserved!
* Random priming - suppliers will generate short oligos of arbitrary sequence.
* Both amplification and priming can lead to biases.

---

# Metabarcoding

* Not the same as "barcoding" as a step in next-generation sequencing to identify templates from different samples.
* A genetic marker (short region of the genome) that should uniquely identify a species.
* It should be universal: found in all organisms in a large taxonomic group.
* There should be alternating regions of high and low sequence conservation.

---

# 16S RNA

<table>
<tr>
  <td>
    <ul>
      <li>16S ribosomal RNA is a component of 30S small subunit of prokaryotic ribosome</li>
      <li>Use of 16S for phylogenetics proposed by Carl Woese and George Fox in [1977](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC432104/).</li>
      <li>Presently the gold standard for metabarcoding studies of bacterial microbiomes.</li>
    </ul>
  </td>
  <td width="40%">
    <img src="https://upload.wikimedia.org/wikipedia/commons/3/3d/10_small_subunit.gif"/>
    <small>
    3d animation of 30S small ribosomal subunit (PDB [1FKA](https://www.rcsb.org/structure/1FKA)).
    </small>
  </td>
</tr>
</table>

---

<!-- Image credit: UCSC Center for Molecular Biology of RNA -->
<section data-background="/img/thermus_16s_2ndry.jpg" width="700px">

---

# Variable regions of 16S

![](/img/16S.png)
<small>
Image credit: Jethro Johnson, https://www.pacb.com/wp-content/uploads/Jethro-Johnson-PacBio-East-Coast-UGM-2017.pdf
</small>

---

# Taxonomic binning

* With the sequence barcode, you now need to look up the species (or higher taxonomic level).
* Not all 16S sequences in Genbank are classified.
* Some sequences are chimeric (PCR artefacts that combine two or more templates).
* Multiple databases maintained by different groups might not agree on the same sequence.

---

# Taxonomic databases

| Database | URL | # Records | Last update |
|----------|-----|-----------|-------------|
| NCBI RefSeq | [https://www.ncbi.nlm.nih.gov/refseq/](https://www.ncbi.nlm.nih.gov/refseq/targetedloci/16S_process/) | 21,073 | 2019-03-12 |
| SILVA | https://www.arb-silva.de/ | 6,073,181 | 2016-09-30 |
| RDP | https://rdp.cme.msu.edu/ | 3,356,809 | 2017-12-13 |
| Greengenes | http://greengenes.lbl.gov | ~1,242,330? | 2011-05-04 |

---

# Further reading

* [Greengenes, a Chimera-Checked 16S rRNA Gene Database and Workbench Compatible with ARB](https://aem.asm.org/content/72/7/5069)
