## Sequence data formats

![](https://imgs.xkcd.com/comics/standards.png)

---

# Why formats?

* Data that has been generated by equipment or retrieved from a database is written to a file in a particular configuration (format).
* A *file format* is a formal set of rules about how data are to be encoded into a file. 
* Otherwise, we cannot guarantee that the computer will retrieve the original data from a file.

---

# Tabular data

* Tables are a fundamental mode of organizing data
* Rows represent observations
* Columns represent variables
* A common export format for databases.

---

An excerpt of a table summarizing infectious disease prevalence in Britain (1965):

| Cases | Eng. & Wales | Grt. Lnd | Scot. | N.Ire. | [Eire](https://en.wikipedia.org/wiki/%C3%89ire) |
|-------|--------------|----------|-------|--------|------|
| Diptheria | 0 | 0 | 1 | 0 |  |
| Dysentry  | 482 | 85 | 136 | 5 | 2 |
| Encephatlis, acute | 2 | 0 |   | 0 |   |
| Enteric fever, typhoid | 3 | 1 | 4 | 1 |   |
| Measles | 3268 | 153 | 37 | 40 | 73 |

British Medical Journal from 1965 ([Graph and Table of Infectious Disease](https://www.ncbi.nlm.nih.gov/pubmed/20790782))

---

# Comma-separated values (CSV)

* A common format for writing tabular data to a text file.
* Each row appears on a separate line.
* Values are separated by a reserved symbol (delimiter)
* If the value contains the delimiter, enclose the value in quotes.

```
Cases,Eng. & Wales,Grt. Lnd,Scot.,N.Ire.,Eire
Diptheria,0,0,1,0,
Dysentry,482,85,136,5,2
"Encephatlis, acute",2,0,,0,,
"Enteric fever, typhoid",3,1,4,1,
Measles,3268,153,37,40,73
```

---

# Tabular data in bioinformatics

* Many file formats in bioinformatics are simply tabular data with comment fields.
* For example, the [variant call format](https://gatk.broadinstitute.org/hc/en-us/articles/360035531692?id=11005) (VCF) for recording sequence variants is a tab-separated table:

```
##fileformat=VCFv4.2
##contig=<ID=20,length=62435964,assembly=B36,md5=f126cdf8a6e0c7f379d618ff66beb2da,species="Homo sapiens",taxonomy=x>
##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">
##FORMAT=<ID=GQ,Number=1,Type=Integer,Description="Genotype Quality">
##FORMAT=<ID=DP,Number=1,Type=Integer,Description="Read Depth">
##FORMAT=<ID=HQ,Number=2,Type=Integer,Description="Haplotype Quality">
#CHROM POS ID REF ALT QUAL FILTER INFO FORMAT NA00001 NA00002 NA00003
20	14370	rs6054257	G	A	29	PASS	NS=3;DP=14;AF=0.5;DB;H2	GT:GQ:DP:HQ 0|0:48:1:51,51	1|0:48:8:51,51	1/1:43:5:.,.
20	17330	.	T	A	3	q10	NS=3;DP=11;AF=0.017	GT:GQ:DP:HQ	0|0:49:3:58,50 0|1:3:5:65,3	0/0:41:3
20	1110696	rs6040355	A	G,T	67	PASS	NS=2;DP=10;AF=0.333,0.667;AA=T;DB GT:GQ:DP:HQ	1|2:21:6:23,27	2|1:2:0:18,2	2/2:35:4
```

---

# Sequence data formats

* Sequence data are more complicated.
* The relative position of a nucleotide or amino acid is significant.  Swapping "columns" results in a completely different data set.
* Different sequence lengths breaks a standard assumption of tabular data.
* We may need to associate sequences with complex metadata.

---

# Nucleotides

* [IUPAC](https://iupac.org/) defines the following symbols to represent nucleotides and their mixtures:

| Symbol | Bases | Symbol | Bases | Symbol | Bases |
|--------|-------|--------|-------|---------|-------|
| W | A,T | B | A,C,T | N | A,C,G,T |
| R | A,G | D | A,G,T | X | A,C,G,T |
| K | G,T | H | A,C,T |
| Y | C,T | V | A,C,G |
| S | C,G |
| M | A,C |

> Can you think of a [mnemonic](https://en.wikipedia.org/wiki/Mnemonic) for 2-fold mixtures?

---

# FASTA

* One of the most common file formats for sequence data.
* Originated from a sequence alignment program ([FAST-All](https://en.wikipedia.org/wiki/FASTA), 1987) that is no longer used.
* Every sequence record starts with a `>` symbol, followed by the sequence label (header).
* The sequence appears on subsequent lines until the next `>`.

```
>KP728283.1 Zaire ebolavirus isolate Ebola virus/H.sapiens-wt/CHE/2014/Makona-GE1, complete genome
GGATCTTTTGTGTGCGAATAACTATGAGGAAGATTAATAATTTTCCTCTCATTGAAATTTATATCGGAAT
TTAAATTGAAATTGTTACTGTAATCATACCTGGTTTGTTTCAGAGCCATATCACCAAGATAGAGAACAAC
```

---

# PHYLIP

* Used by [Joe Felsenstein](http://evolution.genetics.washington.edu/phylip.html)'s phylogenetic software package of the same name
* The first line has two numbers: the number of sequences, and the sequence length
* Sequence labels are limited to 10 characters in length.

```
1  140
KP728283.1 GGATCTTTTGTGTGCGAATAACTATGAGGAAGATTAATAATTTTCCTCTC 
           ATTGAAATTTATATCGGAATTTAAATTGAAATTGTTACTGTAATCATACC
           TGGTTTGTTTCAGAGCCATATCACCAAGATAGAGAACAAC
```

---

# NEXUS

* The [NEXUS format](https://en.wikipedia.org/wiki/Nexus_file) was designed to incorporate many different data types, including sequences.
* Data are organized into *blocks* enclosed by `BEGIN` and `END` tags.
* Used by several phylogenetic software packages.

```
#NEXUS

begin data;
dimensions ntax=1 nchar=140;
format interleave datatype=DNA missing=N gap=-;

matrix
KP728283.1           GGATCTTTTGTGTGCGAATAACTATGAGGAAGATTAATAATTTTCCTCTC
KP728283.1           ATTGAAATTTATATCGGAATTTAAATTGAAATTGTTACTGTAATCATACC
KP728283.1           TGGTTTGTTTCAGAGCCATATCACCAAGATAGAGAACAAC
;

end;
```

---

# FASTQ format

* FASTQ is essentially an extension of the FASTA format to include quality scores.

```
@SRR5261740.1 1 length=295
AAGCAGTGGTATCAACGCAGAGTACATGGGGACAGTGACCCTGATCTGGTAAAGCTC...
+SRR5261740.1 1 length=295
BBBBBFFFBFFFGGGGGGGGGGHHHHHHHHGGGGGHHHHGHHHGHHHHHHHHGHHHH...
```

* A *quality score* is the predicted probability that the base call is incorrect.
* To save space, this probability is transformed into a single character (more on this later!)

---

# Genbank format

* A very complex format that contains a diverse amount of information:

```
LOCUS       KP728283               18926 bp    cRNA    linear   VRL 11-FEB-2015
DEFINITION  Zaire ebolavirus isolate Ebola
            virus/H.sapiens-wt/CHE/2014/Makona-GE1, complete genome.
ACCESSION   KP728283
VERSION     KP728283.1
KEYWORDS    .
SOURCE      Zaire ebolavirus
  ORGANISM  Zaire ebolavirus
            Viruses; ssRNA viruses; ssRNA negative-strand viruses;
            Mononegavirales; Filoviridae; Ebolavirus.
```

```
FEATURES             Location/Qualifiers
     source          1..18926
                     /organism="Zaire ebolavirus"
                     /mol_type="viral cRNA"
                     /isolate="Ebola virus/H.sapiens-wt/CHE/2014/Makona-GE1"
                     /host="Homo sapiens"
                     /db_xref="taxon:186538"
                     /country="Switzerland"
                     /collection_date="21-Nov-2014"
```

---

# Converting between file formats

* One of the fundamental tasks in bioinformatics is the conversion of data from one format to another.
* Different programs write data to files in different formats, even when the data contain the same information.
* Converting formats is often a required step to feed the output of one program as input for another (building *pipelines*).

---

# Online/GUI-based conversion tools

* [EMBOSS Seqret](https://www.ebi.ac.uk/Tools/sfc/emboss_seqret/)
* [LANL HIV Sequence Database](https://www.hiv.lanl.gov/content/sequence/FORMAT_CONVERSION/form.html)
* [Geneious](https://www.geneious.com/features/import-export-sequence-data/)

---

# How do we really do it?

* Web/GUI interfaces not adequate for building an analysis pipeline
* Several open-source programs for converting formats:
  * Bio&ast; libraries (*e.g.*, BioPython.SeqIO module)
  * [seqmagick](https://github.com/fhcrc/seqmagick), essentially a front-end for SeqIO
* but often we just have to do it ourselves -- this is why **scripting languages** are so popular.

---

# How Perl saved the Human Genome Project
An article in *The Perl Journal* by [Lincoln Stein](https://oicr.on.ca/investigators/lincoln-stein/)

* Project start date 1990, involving many groups.
* Estimated 1 to 10 terabytes needed to complete project.
* Different groups came up with different data exchange formats.
* Perl enabled different groups to rapidly convert outputs to the other group's format.

---

# How Perl saved the Human Genome Project (2)

> Despite the fact that everyone was working on the same problems, no two groups took exactly the same approach. Programs to solve a given problem were written and rewritten multiple times. While a given piece of software wasn't guaranteed to work better than its counterpart developed elsewhere, you could always count on it to sport its own idiosyncratic user interface and data format. 

Lincoln Stein (1996)

---

# Perl is a scripting language

* "Scripting language" is difficult to define.
* Generally a script is a program that automates the execution of tasks.
* A scripting language is often developed using a compiled language - it operates at a higher (more abstract) level.
* (Like making macaroni and cheese with a Kraft Dinner mix instead of growing a wheat field, harvesting the grain, milling the grain into flour...)

---

# Perl was the backbone of bioinformatics

* Perl (Practical Extraction and Reporting Language) was developed by Larry Wall (first release 1987)
* The *lingua franca* of bioinformatics for many years.
* Reputation for enabling developers to do work quickly ("there's more than one way to do it"), but can be difficult to read.

---

A bit of Perl code by Illumina to find genome positions with an `N`

```
my $a=();
my $start=0;
#====finding Ns in the specified genomic range=====
open(IN, "$ARGV[0]"); #genome fasta
my @lines = <IN>;
close IN;
my $pos = $start;
my %Ns = ();
for($i=1;$i<@lines;$i++) #The 1st line is header
{
  chomp $lines[$i];
  @a = split //,$lines[$i];
  for($j=0;$j<@a;$j++) {
    if($a[$j] eq 'N') {
      $Ns{$pos} = 1;
    }
    $pos++;
  }
}
```

---

# Python

* Another scripting language developed by Guido van Rossum (first release 1990).
* "There should be one - and preferably only one - obvious way to do it."
* Notorious for whitespace requirements ("Readability counts.").
* Has overtaken Perl in popularity, even bioinformatics.
* Both Perl and Python have [Bio&ast;.SeqIO](https://biopython.org/wiki/SeqIO) libraries for converting sequence formats.

---

A conversion of the Illumina Perl code to Python
```python
import sys

# read filename from command line
infile = sys.argv[1]

handle = open(infile, 'rU')
_ = handle.readline()  # discard header line

Ns = []  # span multiple lines
for line in handle:
    # we assume that the line consists entirely of nucleotide sequence
    seq = line.strip('\n')
    for base in seq:
        Ns.append(1 if base == 'N' else 0)
```

---

# Suggested readings

* [How Perl Saved the Human Genome Project](https://web.archive.org/web/20070202101624/http://www.bioperl.org/wiki/How_Perl_saved_human_genome)



