# MIMM4750G
## de novo assembly
![](https://imgs.xkcd.com/comics/word_puzzles.png)

---

# What is *de novo* assembly?

* Combine short reads that seem to overlap so they form a longer sequence.
* A *contig* is a contiguous (uninterrupted) run of nucleotides that is formed from the assembly of short reads.

<!--* Some methods attempt to further assemble contigs into *scaffolds*.-->
<img src="/img/contig.png" width="500px"/>

---

# Pros and cons

* Short-read *mapping* is generally faster and easier than *de novo* assembly, but needs a good reference.
* Mapping better suited for variant detection.
* *de novo* assembly is better suited for discovering new genomes, where no suitable reference exists.
* Hybrid methods use both *de novo* assembly and mapping to re-assemble local contigs.

---

# *de novo* assembly for pathogens

Percentage of unique reads as a function of read length for (a) $\lambda$-phage and (b) E.coli K12. 

![](/img/gni170f1.jpeg)

<small>
Figure from N Whiteford *et al.* (2005) Nucleic Acids Research 33: e171.
</small>

---

# Finding overlaps

* This has been compared to opening a box of ten million puzzle pieces and finding which pieces match together...
* except that the edges always line up, and some of the pieces are repeated many times, and other pieces belong to different sets...


<img src="/img/jigsaw.jpg" width="400px"/>

<small><small><small>
Image source: http://homeli.co.uk/1000-colours-rainbow-cmyk-gamut-jigsaw-puzzle-by-clemens-habicht/ 
</small></small></small>

---

# Finding overlaps

<img src="/img/contig2.png" width="500px"/>


---

# Further readings

* [De novo assembly of short sequence reads](https://academic.oup.com/bib/article/11/5/457/1746253) by K Paszkiewicz and DJ Studholme (2010), Briefings in Bioinformatics 11

