# MIMM4750G
## SIR and the BEAST

![](https://imgs.xkcd.com/comics/outbreak.png)

---

# Models of epidemics

* Suppose a single infected individual enters a population.
* Let $I$ be the number of infected people.
* Let $S$ be the number of uninfected people susceptible to infection.
* Let $\beta$ be the rate that a susceptible person who comes in contact with an infected person becomes infected.
* The rate of new infections is $\beta S I = \beta (N-I)I$.

---

# The dumb assumptions

1. The population is "well mixed".  Any one person is equally likely to come into contact with any other person.

2. The transmission rate $\beta$ is the same for everybody, *forever*.

3. The size of the total population does not change.

---

# SI dynamics

* This model is known as the susceptible-infected (SI) model.
* $I$ increases at the fastest rate when $S=I$, *i.e.*, $I=N/2$.
![](/img/SI-animate.gif)

---

# Compartmental models

* The SI model belongs to a class of epidemic models known as [compartmental models](https://en.wikipedia.org/wiki/Compartmental_models_in_epidemiology).
* A compartment is a sub-population with distinct numbers and rate parameters.
* These models are generally represented by systems of differential equations.
* For example, the SI model can be written as:

$$\frac{dS}{dt}=-\beta S I, \; \frac{dI}{dt} = \beta S I$$

---

# Compartmental models

* SIS: infected individuals recover and become susceptible again.
* SIR: adds a "removed" compartment where infected individuals eventually *recover* or *die*.
* SEIR: individuals become *exposed* but not immediately infectious.
* and many others!

---

# $R_0$

* "R naught" (basic reproduction number) is a fundamental parameter of epidemiology.
* The expected number of secondary cases from the index case.
* If $R_0<1$, then we expect the infection to die out.  If $R_0>1$, then we expect it to spread.
* For the SIR model (with death rate $\gamma$), 

  $$R_0 = \frac{\beta N}{\gamma}$$
  
* Increases with transmission rate ($\beta$), and declines with death rate $\gamma$.

---

# Birth-death models

* The transmission of a pathogen to a new host is a *lineage birth*.
* The removal of an infected individual by recovery or death is a *lineage death*.
* If we follow these lineages forward in time, we can generate a tree.

<img src="/img/birth-death.png" width="400px"/>

---

# Coalescent and birth-death

* Both are models of how trees are shaped by population-level processes.

| Coalescent | Birth-death |
|------------|-------------|
| Reverse time | Forward time |
| Assumes $n<<N$ | Robust to large $n$ |
| Sampling times are data | Sampling must be modeled |
| No selection | Can handle selection naturally |
| Better for a single population | Better for pathogens? |

* Both models are *priors* - if you have enough data, it shouldn't really matter which one you use.

---

# BEAST

* Bayesian Evolutionary Analysis by Sampling Trees
* Primarily developed by Alexei Drummond and Andrew Rambaut in the [Java](https://en.wikipedia.org/wiki/Java_(programming_language) programming language.
* BEAST has become one of the most influential software packages in infectious disease research in the last decade.
* Over 10,000 citations, including over 40 Nature papers and over 20 Science papers since 2007.

---

# BEAST

* Can use the coalescent or birth-death model as a prior over trees.
* Rearranges and rescales parts of the tree to perform a *random walk* over the posterior distribution of trees.
* At the same time, samples parameters of a model of evolution that is needed to reconstruct the tree from sequences.
* Outputs trace logs of posterior, likelihood, prior and model parameters &mdash; trees are written to a separate log.

---

# XML

* A BEAST analysis is set up in an XML file
* eXtensible Markup Language

```xml
<?xml version="1.0"?>
<beast>
	<alignment>
		<sequence dataType="nucleotide">
			<taxon id="Brazi82"/>
			ATGATCGTAGTATCGTAGCTCGGTTTTTACGATCGGAC
		</sequence>
		<sequence dataType="nucleotide">
			<taxon id="ElSal83"/>
			ATGATCGTAGTATCGTAGCTCGGTTTTTACGATCGGAC
		</sequence>
	</alignment>
</beast>
```

---

# Generating XML

* Most users set up an analysis with the GUI program BEAUti (*Bayesian Evolutionary Analysis Utility*) rather than directly edit XML.

<img src="/img/beauti.png" width="600px"/>

---

# Running BEAST

* BEAST now requires [BEAGLE](https://github.com/beagle-dev/beagle-lib) to run.
* "Broad-platform Evolutionary Analysis General Likelihood Evaluator".
* BEAGLE is a [library](https://en.wikipedia.org/wiki/Library_(computing) &mdash; a collection of functions that can be called from other programs.
* Using BEAGLE can make BEAST about 5-10 times faster!
* (It used to be possible to run BEAST *without* BEAGLE...)

---

# Running BEAST

* BEAST can take a very long time! 
* Since we are trying to explore an enormous model space, we usually run chains of $10^8$ steps or more!
* BEAST tries to estimate how long it takes to perform 1 million steps.

```
# BEAST v1.10.5 Prerelease #23570d1
# Generated Sun Mar 03 13:42:57 EST 2019 [seed=1551638575176]
# benchmark2.xml
state	Posterior   	Prior       	Likelihood  	rootHeight  	clock.rate  	alpha       
0	-568698.9433	-16.9645    	-568681.9788	0.20000     	1.00000     	0.50000     	-
10000	-215056.1166	59.1339     	-215115.2505	0.26252     	1.00000     	0.32615     	-
20000	-202604.8926	72.8223     	-202677.7148	0.23844     	1.00000     	0.30842     	0.11 hours/million states
30000	-196143.5414	72.3079     	-196215.8493	0.27209     	1.00000     	0.26344     	0.11 hours/million states
40000	-193720.6293	59.7853     	-193780.4145	0.39626     	1.00000     	0.26003     	0.11 hours/million states
50000	-193206.4970	56.6362     	-193263.1333	0.45645     	1.00000     	0.24143     	0.11 hours/million states
```

---

# Ebolavirus outbreak in West Africa

![](/img/gire-ebola.svg)

<small>
SK Gire *et al.* 2014. Genomic surveillance elucidates Ebola virus origin and transmission during the 2014 outbreak. *Science* 345: 1369-1372.
</small>

---

# Origin of modern avian influenza virus

<img src="/img/worobey-influenza.svg" width="700px"/>

<small>
M Worobey, G-Z Han and A Rambaut, 2014. A synchronized global sweep of the internal genes of modern avian influenza virus. Nature 508: 254-257.
</small>

---

# Neolithic expansion of human tuberculosis

<img src="/img/comas-MTBC.png" width="700px"/>

<small>
I Comas *et al.* 2013. Out-of-Africa migration and Neolithic co-expansion of
*Mycobacterium tuberculosis* with modern humans. Nature Genetics 45: 1176-1182.
</small>
