# MBI/MIMM 4750G
## Introduction and review

![](https://imgs.xkcd.com/comics/standards.png)

---

# Schedule

* Lectures on Mondays and Fridays at 9:30am
* Lab practicals on Wednesdays from 9:30am to 11:30am in the Arthur and Sonia Labatt Health Science Building, Genlab HSB-16.

<img src="/img/HSB-map.png" height="300px"/>

---

# Learning objectives

* Assemble genomes of unknown viruses from [metagenomic](https://en.wikipedia.org/wiki/Metagenomics) data.
* Cluster virus sequences for [classification](https://en.wikipedia.org/wiki/Virus_classification) and outbreak detection.
* Understand and apply different strategies for [tree-building](https://en.wikipedia.org/wiki/Computational_phylogenetics).
* Detect sites under [positive selection](https://en.wikipedia.org/wiki/Directional_selection) in a virus genome.
* Use [Bayesian methods](https://en.wikipedia.org/wiki/Bayesian_inference) to fit epidemic models to dated sequences.
* Understand the [ethical challenges](https://en.wikipedia.org/wiki/Information_privacy) of bioinformatics and data sharing.

---

# Evaluation

* This is an essay course!
* Three components:
  * Computing lab assignments - 50%
  * Project proposal (due February 21, 2022) - 15%
  * Project final report (due April 1, 2022) - 35%

* Turnitin *will* be used to assess written assignments.

---

# How to do well

* Labs:
  * This course has no prerequisites in computer science - you will not have to write any of your own code.
  * *However*, you will need to become comfortable with the command line interface.

* Writing:
  * Start writing your methods section first - it is the easiest to write.
  * Next, go into your results by describing your analysis outputs.

---

# Data formats

* Data that has been generated by equipment or retrieved from a database is written to a file in a particular configuration (format).
* A *file format* is a formal set of rules about how data are to be encoded into a file.
  * Enables a computer to systematically read and write data to and from a file.
* Without these rules, we cannot guarantee that the computer will retrieve the original data from a file.

---

# Tabular data

* Tables are a fundamental mode of organizing data
* Rows represent observations
* Columns represent variables
* A common export format for databases.

---

# Tabular data in bioinformatics

* Many file formats in bioinformatics are simply tabular data with comment fields.
* For example, the [variant call format](https://gatk.broadinstitute.org/hc/en-us/articles/360035531692?id=11005) (VCF) for recording sequence variants is a tab-separated table:

```
##fileformat=VCFv4.2
##contig=<ID=20,length=62435964,assembly=B36,md5=f126cdf8a6e0c7f379d618ff66beb2da,species="Homo sapiens",taxonomy=x>
##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">
##FORMAT=<ID=GQ,Number=1,Type=Integer,Description="Genotype Quality">
##FORMAT=<ID=DP,Number=1,Type=Integer,Description="Read Depth">
##FORMAT=<ID=HQ,Number=2,Type=Integer,Description="Haplotype Quality">
#CHROM POS ID REF ALT QUAL FILTER INFO FORMAT NA00001 NA00002 NA00003
20	14370	rs6054257	G	A	29	PASS	NS=3;DP=14;AF=0.5;DB;H2	GT:GQ:DP:HQ 0|0:48:1:51,51	1|0:48:8:51,51	1/1:43:5:.,.
20	17330	.	T	A	3	q10	NS=3;DP=11;AF=0.017	GT:GQ:DP:HQ	0|0:49:3:58,50 0|1:3:5:65,3	0/0:41:3
20	1110696	rs6040355	A	G,T	67	PASS	NS=2;DP=10;AF=0.333,0.667;AA=T;DB GT:GQ:DP:HQ	1|2:21:6:23,27	2|1:2:0:18,2	2/2:35:4
```

---

# Sequence data formats

* Sequence data are more complicated.
* The relative position of a nucleotide or amino acid is significant.  Swapping "columns" results in a completely different data set.
* Different sequence lengths breaks a standard assumption of tabular data (although aligned sequences are essentially tabular).
* We may need to associate sequences with complex metadata.

---

# Nucleotides

* [IUPAC](https://iupac.org/) defines the following symbols to represent nucleotides and their mixtures:

| Symbol | Bases | Symbol | Bases | Symbol | Bases |
|--------|-------|--------|-------|---------|-------|
| W | A,T | B | C,G,T | N | A,C,G,T |
| R | A,G | D | A,G,T | X | A,C,G,T |
| K | G,T | H | A,C,T |||
| Y | C,T | V | A,C,G |||
| S | C,G |||||
| M | A,C |||||

---

# FASTA

* One of the most common file formats for sequence data.
* Originated from a sequence alignment program ([FAST-All](https://en.wikipedia.org/wiki/FASTA), 1987) that is no longer used.
* Every sequence record starts with a `>` symbol, followed by the sequence label (header).
* The sequence appears on subsequent lines until the next `>`.

```
>NC_045512.2 Severe acute respiratory syndrome coronavirus 2 isolate Wuhan-Hu-1, complete genome
ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAA
CGAACTTTAAAATCTGTGTGGCTGTCACTCGGCTGCATGCTTAGTGCACTCACGCAGTATAATTAATAAC
TAATTACTGTCGTTGACAGGACACGAGTAACTCGTCTATCTTCTGCAGGCTGCTTACGGTTTCGTCCGTG
```

---

# Genbank format

* A very complex format that contains a diverse amount of information:

```
LOCUS       NC_045512              29903 bp ss-RNA     linear   VRL 18-JUL-2020
DEFINITION  Severe acute respiratory syndrome coronavirus 2 isolate Wuhan-Hu-1,
            complete genome.
ACCESSION   NC_045512
```

```
FEATURES             Location/Qualifiers
     source          1..29903
                     /organism="Severe acute respiratory syndrome coronavirus
                     2"
                     /mol_type="genomic RNA"
                     /isolate="Wuhan-Hu-1"
```

---

# Converting between file formats

* One of the fundamental tasks in bioinformatics is the conversion of data from one format to another.
* Different programs write data to files in different formats, even when the data contain the same information.
* Converting formats is often a required step to feed the output of one program as input for another (building *pipelines*).

---

# Sequence alignment

* An alignment is a hypothesis about how residues (nt, aa) in homologous sequences are related to residues in a common ancestor.
* This is not trivial because of [insertions](https://en.wikipedia.org/wiki/Insertion_(genetics)) and [deletions](https://en.wikipedia.org/wiki/Deletion_(genetics)).

```
Query  1    CTRPNNTRKSVSITIGPGRASYATG---GQAHC  30
            ||||||||||  |||||||||||||   |||||
Sbjct  95   CTRPNNTRKS--ITIGPGRASYATGGIIGQAHC  125

```

---

Unaligned HCV sequences

![](/img/unaligned-hcv.png)

---

Aligned HCV sequences

![](/img/aligned-hcv.png)

---

# Gap characters

* The presence of an insertion or deletion is indicated by a gap character.
* By convention, we use a single dash "&ndash;" for each indel.
* Some programs use non-standard characters like "`.`", "`~`" or "`X`".
* Without additional information, we cannot tell whether a gap is the result of an insertion in the longer sequence or a deletion in the shorter.
* Hence we use the [portmanteau](https://en.wiktionary.org/wiki/portmanteau) *indel* (insertion/deletion).

---

# Score matrices

* A major feature of heuristic methods of alignment
* A score quantifies the likelihood of a residue being replaced by another.
  * Find which alignment of two sequences maximizes the score.
* A simple score matrix for nucleotides: $+1$ (match), $-1$ (mismatch).

|   | A | C | G | T |
|---|---|---|---|---|
| A | +1 | -1 | -1 | -1 |
| C | -1 | +1 | -1 | -1 |
| G | -1 | -1 | +1 | -1 |
| T | -1 | -1 | -1 | +1 |

---

# Gap penalties

* We need to penalize the score for gaps, or else an alignment gets gaps for free:
  ```
  A-C-G-T   ACGT
  -A-C-T-   AC-T
  ```
* The left option is obviously a terrible alignment!
* If we use match/mismatch scores of `+1/-1` and a gap penalty of `-1`, then **what are the scores** for these alignments?

---

# Penalizing longer gaps

* If a gap spans 2 or more residues, we might want to enforce a milder or more severe penalty.
* For a gap of length $l$:
  * **Linear gap penalty** = $-ld$, where $d$ is a constant per-gap penalty.
  * **Affine gap penalty** = $-d-(l-1)e$, where $d$ is the gap *opening* penalty and $e$ is the gap *extension* penalty.
* It is more common to use [affine](https://en.wikipedia.org/wiki/Affine_combination) gap penalties.

---

# Terminal gaps

* Terminal gaps are a contiguous run of gaps on either extreme left or right of a pairwise alignment.
* Also known as "leading" and "trailing" gaps.

  ```
  ACTGATC   ACTGATC
  ---GATC   ACTG---
  ```

* We might not want to penalize these when aligning partial (incomplete) sequences.

---

# Multiple sequence alignment

* It is not trivial to extend pairwise alignment to more than two sequences.
  ```
  1 AC-GT   1 ACGT   1 ACGT
    || ||     | ||   2 A-GT
  3 ACAGT   2 A-GT   3 ACAGT
  ```
* We use a *guide tree* to propagate pairwise alignments to all sequences.
  * To start, we need a way to build a tree that does not need an alignment!
  * The accuracy of alignment can be improved by building a tree from the alignment output, and plugging it back in as a better guide tree.
