# MIMM4750G
## Metabarcoding

![](https://imgs.xkcd.com/comics/hofstadter.png)

---

# What do we do with our NGS data?

* So far we've talked about applications of NGS that focus on a single species (bacteria or virus).
* The massive throughput of NGS platforms allows us to target multiple species, or perform untargeted sequencing.
* A key challenge to untargeted sequencing is amplification.
* Amplification requires primers, but primers are usually specific to a particular sequence.

---

# Metabarcoding

* Not the same as "barcoding" as a step in next-generation sequencing to identify templates from different samples.
  * *a.k.a.* targeted metagenomics, amplicon-based metagenomics, and metagenetics.
* A genetic marker (short region of the genome) that should uniquely identify a species.
  * It should be universal: found in all organisms in a large taxonomic group.
  * There should be alternating regions of high and low sequence conservation.
* More useful for bacteria than viruses!

---

# 16S RNA

<table>
<tr>
  <td style="font-size: 26px">
    <ul>
      <li>16S ribosomal RNA is a component of 30S small subunit of prokaryotic ribosome</li>
      <li>Use of 16S for phylogenetics proposed by <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC432104/">Carl Woese and George Fox</a> in 1977.</li>
      <li>Presently the gold standard for metabarcoding studies of bacterial microbiomes.</li>
    </ul>
  </td>
  <td width="40%">
    <img src="https://upload.wikimedia.org/wikipedia/commons/9/90/010_small_subunit-1FKA.gif"/>
    <small>
    3d animation of 30S small ribosomal subunit (PDB [1FKA](https://www.rcsb.org/structure/1FKA)).
    </small>
  </td>
</tr>
</table>

---

<small> Image credit: [UCSC Center for Molecular Biology of RNA](http://rna.ucsc.edu/rnacenter/images/figs/thermus_16s_2ndry.pdf)</small>
<!-- <section data-background="/img/thermus_16s_2ndry.svg"/> -->
<img src="/img/thermus_16s_2ndry.svg" width="800px"/>

---

# Variable regions of 16S

<img src="/img/16S.png" width="700px"/>
<small>
Image credit: Jethro Johnson, https://www.pacb.com/wp-content/uploads/Jethro-Johnson-PacBio-East-Coast-UGM-2017.pdf
</small>

---

# OTU clustering

* In fact, we don't make taxonomic assignments to sequences on a one-to-one basis.
* [Operational Taxonomic Units](https://en.wikipedia.org/wiki/Operational_taxonomic_unit): clusters of sequences based on genetic similarity.
* Hence, we need some [genetic distance](http://slides.filogeneti.ca/html/mimm4750g-L06-distance.html#/6) measure.
* 97% sequence identity has become a standard cutoff.
* Clustering saves a lot of computing!

---

# DOTUR

* Defining Operational Taxonomic Units and estimating species Richness ([Schloss and Handelsman](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1065144/)).
  * Nearest-neighbor clustering: an [agglomerative method](http://slides.filogeneti.ca/html/mimm4750g-L07-clusters.html#/5) where a sequence joins a cluster if it its similarity to *any* member is below a threshold.
  * Farthest-neighbor: sequence must have similarity below threshold for *every* member of the cluster.
  * UPGMA: replace clustering pair with average of their distance to other sequences.
* Recall that UPGMA was [a distance-based method for building trees](http://slides.filogeneti.ca/html/mimm4750g-L08-trees.html#/12).

---

# Dealing with sequencing error

* Sequences may be different because of either within-species variation or sequencing error.
* [DADA2](https://benjjneb.github.io/dada2) fits an error model while clustering reads.
* All identical reads are grouped into *variants*, tracking read abundance.
  * All variants start in <a href="http://slides.filogeneti.ca/html/mimm4750g-L07-clusters.html#/5">one big cluster</a>.
  * Every variant is compared to most abundant member of its cluster.
  * Let $p$ be the probability that a variant with abundance $n$ was produced by sequencing errors from the most abundant member.
  * If $p$ is too low, variant becomes a new cluster.  Repeat until no new clusters.

---

# Chimera detection

<table>
<tr>
  <td>
  <ul>
  <li><a href="https://en.wikipedia.org/wiki/Chimera_(EST)">Chimeras</a> are PCR artifacts produced when an incomplete extension anneals to another template to prime the next reaction.</li>
  <li>For example, <a href="http://www.drive5.com/usearch/manual/uchime_algo.html"/>UCHIME</a> (Robert Edgar) splits query sequence into four parts.</li>
  <li>Each part is mapped to a reference database assumed to be chimera-free.</li>
  <li>Retrieves two potential "parent" sequences.</li>
  <li>See also <a href="http://microbiomeutil.sourceforge.net/">ChimeraSlayer</a>, <a href="https://www.ncbi.nlm.nih.gov/pubmed/21276213">AmpliconNoise Perseus</a></li>
  </ul>
  </td>
  <td width="40%">
    <img src="/img/btr381.png" width="300px"/>
    <small>
      Image credit: R Knight (2011) Bioinformatics 27: 2195. doi:10.1093/bioinformatics/btr381
    </small>
  </td>
</tr>
</table>


---

# Taxonomic assignment

* With the sequence barcode, you now need to look up the species (or higher taxonomic level).
* Not all 16S sequences in Genbank are classified.
* Some database sequences are chimeric (PCR artefacts that combine two or more templates).
* Multiple databases maintained by different groups might not agree on the same sequence.

---

# Taxonomic databases

| Database | URL | # Records | Last update |
|----------|-----|-----------|-------------|
| NCBI RefSeq | [https://www.ncbi.nlm.nih.gov/refseq/](https://www.ncbi.nlm.nih.gov/refseq/targetedloci/16S_process/) | 21,442 | 2020-02-20 |
| SILVA SSU (138) | https://www.arb-silva.de/ | 2,225,272| 2019-12-16 |
| RDP (release 11) | https://rdp.cme.msu.edu/ | 3,356,809 | 2016-09 |
| Greengenes | http://greengenes.lbl.gov | 1,262,986 | 2015-05 |

---

# Who do you believe?

>  I found 249,490 identical sequences with conflicting annotations in SILVA v128 and Greengenes v13.5 at ranks up to phylum (7,804 conflicts), indicating that the annotation error rate in these databases is ~17%.

<small>
Robert C. Edgar (developer of *MUSCLE*), 2018. *Taxonomy annotation and guide tree errors in 16S rRNA databases*, PeerJ: 5030.
</small>

* Some sequences actually from organelles (*e.g.*, mitochondria) or eukaryotes (*e.g.*, protists; [Lesack and Birol 2018](https://www.biorxiv.org/content/10.1101/441576v1.abstract)).

---

# Methods for taxonomic assignment

* After choosing a reference database, you need a method to assign taxa.
* [BLAST](http://slides.filogeneti.ca/html/mimm4750g-L04-BLAST.html#/)-based tools:
  * [MEGAN](https://uni-tuebingen.de/fakultaeten/mathematisch-naturwissenschaftliche-fakultaet/fachbereiche/informatik/lehrstuehle/algorithms-in-bioinformatics/software/megan6/) - post-processing of BLAST results to assign lowest common ancestors.
* [USEARCH](https://www.drive5.com/usearch/) (R.Edgar again)
  * Closed source, feature-limited distribution.
  * Measures the intersection for the sets of all 8-mers in query sequence and database sequence.

---

# Naive Bayes classifier

* [RDP Classifier](https://github.com/rdpstaff/classifier): open source, implemented in Java.
* Also extracts 8-mers from query and database sequences.
* The probability that sequence $S$ is a member of genus $G$ is:

  $$P(S | G)= \prod_{w_i\in S} \frac{n_G(w_i) + P(w_i)}{N_G+1}$$

  * $n_G$ is the number of sequences in $G$ with word $w_i$
  * $N$ is the total size of $G$
  * $P(w_i)$ is the prior probability (the overall frequency of $w_i$).

---

# Phylogenetic placement

* Another approach to taxonomic assignmermnt is to place a new sequence in a fixed reference tree:
  * [pplacer](https://github.com/matsen/pplacer) - by Matsen group, no longer maintained.
  * [EPA](https://github.com/stamatak/standard-RAxML) - part of RAxML
* Much faster than regenerating the entire tree!
* Uses a [model of evolution](http://slides.filogeneti.ca/html/mimm4750g-L10-rates.html#/8), better measure of distance.
* Used to estimate that about 1% of 16S rRNA databases are misclassified.

---

# Pipelines

* A pipeline or workflow is a collection of programs and scripts used to process data in a fixed sequence.
* Pipelines for metabarcoding have become extremely popular.

* [mothur](https://github.com/mothur/mothur) - open source C++, all in-house programs

* [QIIME](https://github.com/qiime2) - Python scripts that wrap around a collection of third-party programs.

---

# Oral pathogen coinfections

* [*Aggregatibacter actinomycetemcomitans*](https://en.wikipedia.org/wiki/Aggregatibacter_actinomycetemcomitans) is a major cause of destructive gum disease.
* 16S metabarcoding revealed that *A. actinomycetemcomitans* is more abundant in association with non-oral microbes.

<img src="/img/19685-f1.png" width="650px"/>

<small>Image credit: GR Lewin <i>et al.</i> (2019) PNAS 116: 19685. https://doi.org/10.1073/pnas.1907619116</small>

---

# Exposure to antibiotics disrupts bee gut microbiome

* Tetracyclines induce loss of gut microbiota, altered composition.
* Infection by [*Serratia*](https://en.wikipedia.org/wiki/Serratia) bacteria induces greater mortality in tetracycline-exposed bees.

<img src="/img/bees.png" width="700px"/>

<small>Image credit: K Raymann <i>et al.</i> (2017) PLOS Biology 15: e2001861. https://doi.org/10.1371/journal.pbio.2001861</small>

---

# Altered bacterial microbiome in HIV patients

* HIV-infected Ugandans with low CD4 cell counts had less diverse gut bacterial microbiota.
* Increased abundance of enterobacteria associated with inflammation.
<img src="/img/hiv-microbiome.jpg" width="750px"/>
<small>Image credit: CL Monaco <i>et al.</i> (2016) Cell Host & Microbe 19: 311. https://doi.org/10.1016/j.chom.2016.02.011</small>

---

# Further reading

* [Taxonomy annotation and guide tree errors in 16S rRNA databases](https://www.biorxiv.org/content/10.1101/288654v2)

* [A perspective on 16S rRNA operational taxonomic unit clustering using sequence similarity](https://www.nature.com/articles/npjbiofilms20164)

* [Analytical Tools and Databases for Metagenomics in the Next-Generation Sequencing Era](https://genominfo.org/journal/view.php?doi=10.5808/GI.2013.11.3.102)

* [Introducing DOTUR, a Computer Program for Defining Operational Taxonomic Units and Estimating Species Richness](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1065144/)

* [Denoising the Denoisers: an independent evaluation of microbiome sequence error-correction approaches](https://peerj.com/articles/5364/)
