# MIMM4750G
## Detecting selection

---

# Natural selection

* Variation in fitness that is associated with variation in the environment.
* Selection is responsible for the spread of drug resistance in pathogens.
* Host adaptation: a mutant allele of the Duffy blood-group antigen (Fy) that reduces risk of infection by *Plasmodium vivax* (vivax malaria) is [near fixation in sub-Saharan Africa](https://www.nature.com/articles/nrg3734).
<img src="/img/vivax.jpg" width="500px"/>

---

# Types of selection

* **Purifying selection:** once the favored genetic variant has been "fixed" in the population, selection continues to remove all other variants.
* Most sites are under purifying selection ("if it ain't broke..")

* **Directional selection:** a specific genetic variant has a selective advantage and increases in frequency.
* Purifying selection is the end result of directional selection.
* Directional selection is difficult to observe; you must be at the right place at the right time.

---

# Diversifying selection

* **Diversifying selection:** different genetic variants are favored in different environments.
* Directional selection depletes genetic variation; diversifying selection promotes variation.
* What causes diversifying selection in pathogens?
* Transmission of an infection from one host environment to another (host-specific immune responses).
* *e.g.,* strong div. selection at cytotoxic T-lymphocyte epitopes in virus proteins.

---

# How do detect selection?

* Longitudinal data: track the frequencies of an allele over time.
* What if we don't know which allele is under selection?  What if we don't have longitudinal data?
* Cross-sectional (comparative) methods: infer selection by comparing genetic sequences sampled at about the same time.
* More convenient, potentially more powerful.

---

# Protein evolution

* Infer selection by comparting relative rates of evolution.
* Requires a baseline/point of reference, *e.g.,* "neutrally evolving" pseudogenes.
* A popular contrast is non-synonymous versus synonymous variation within protein-coding sequence.
* Non-synonymous = nucleotide substitution alters the amino acid encoded by the codon.
* Such approaches are generally called "dN/dS" methods.

---

# dN/dS

* There are 9 possible nucleotide substitutions in a 3-nt codon.
* We assume that "nonsense" substitutions to a stop codon don't persist.
* The genetic code determines how many of these 9 nt changes would result in a non-synonymous change &mdash; this is the *expected* number.
* dN is the ratio between the observed and expected numbers of non-synonymous substitutions.
* dS is the same ratio for synonymous substitutions.

---

# A simple example

* The codon `ACG` encodes threonine (T).  It has 3 synonymous and 6 nonsynymous substitutions (*e.g.*, `ATG` for methionine).
* Suppose we count 10 non-synonymous and 4 synonymous substitutions in the tree.
* We observed over twice as many non-synonymous substitutions!  Is this evidence of strong diversifying selection at this codon?
* 




































