# MIMM4750G
## Short read mapping

![](https://imgs.xkcd.com/comics/making_hash_browns.png)

---

# Two problems of alignment

* When we've been aligning sequences, we've been starting with a set of short, homologous sequences - the problem is figuring out where there might be insertions and deletions.
* What if we need to align a short sequence to a very long one, *e.g., a genome?  
* We have to solve this *mapping* problem (where is the homologous region, if any?) before we can tackle the other one.
* The computing time of pairwise alignment is about $O(mn)$ where $m$ and $n$ are the sequence lengths.

![](/img/lost-read.svg)

---

# NGS and alignment

* The development of NGS platforms created a huge challenge for existing alignment methods &mdash; too much data!
* New alignment programs were needed.
* In 2012, there were over 60 different programs available.
* Hash functions played an important role in many of these methods

---

<img src="/img/bts605f1p.jpeg" width="600px"/>

<small>
Timeline of mapper software from NA Fonseca *et al* 2012, Bioinformatics 28: 3169.
</small>

---

# Hash functions

* A hash function is some algorithm that can take a datum of any size (usually enormous) and quickly reduce it down to a value of a much smaller, fixed size.
* Since the range of possible hash *values* is much smaller than the range of possible *keys*, there is some chance that two or more keys will *collide* to the same value.
* For example, a function that determines if a string has an even or odd number of characters is a hash function (but a pretty useless one).

---

# Hash functions

* MD5 is a hash function that is widely used to verify the contents of a file that has been transferred over the network.
* MD5 produces a 128-bit hash value.  This is usually represented as a hexadecimal string of length 32.  A hexadecimal value (`0` to `f`) represents four bits (*e.g.*, `1101` equals `d`).
* Chromosome 21 of the human genome (assembly hg38) has the MD5 hash value `eefe014d35decf90afde7b37e9954554`.

---

# Hash indexing

* A hash value can be used to solve the first problem of alignment.
1. Break the genome down into fragments.
2. Generate the hash values for each fragment.
3. Use the hash value as the lookup key for the location of that fragment in the genome.
4. We can then rapidly "look up" the location of a matching fragment from our query sequence using this index.

---

# Burrows-Wheeler transform

* Storing the hash index required a lot of memory!
* The next generation of short read mappers (BWA, Bowtie, SOAPv2) replace the hash function with the Burrows-Wheeler transform (BWT)

<img src="/img/btp324f2.jpeg" width="400px"/>

---

# BWT

* The transform of string $S$, $BWT(S)$, is easier to compress
* Reversing $BWT(S)$ back to $S$ is fast.
* Transform makes finding exact matches to another string $T$ *really* fast.

---


