# MIMM4750G
## NGS QC and open source
![](https://imgs.xkcd.com/comics/open_source.png)

---

# Now you have data

* Congratulations!  You have several hundred gigabytes of data.
* Before you start to learn how to *analyze* the data, you need to check if it is any good.
* Quality control is a necessary and tedious step of NGS analysis.
* We will focus on Illumina sequencing - there are many other platforms but right now Illumina is fairly popular.

---

# Demultiplexing

* One of the first steps in processing raw NGS outputs
* Generates FASTQ from base call `.bcl` files
* This conversion used to be performed with a Perl script `bcl2fastq.pl`
* Has now been re-implemented as a C++ program `bcl2fastq2`
* Separates reads labelled with different index tags into different FASTQ files.

---

# Quality control

* **Number of reads:** a sample may have a small number of reads, possibly due to inaccurate DNA quantification
* **Quality scores:** read quality tends to fall off over cycles.
* **Nucleotide frequencies:** skewed frequencies can reflect poor quality.

---

# QC Software

* Several different software packages are available for screening FASTQ data for quality; for example:

<table>
  <tr>
    <td>
    [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
    <img src="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc.png" width="400px"/>
    </td>
    <td>
      [fastx_toolkit](http://hannonlab.cshl.edu/fastx_toolkit/index.html)
      <img src="http://hannonlab.cshl.edu/fastx_toolkit/bc54_nuc.png" width="400px"/>
    </td>
  </tr>

</table>

---

# Trimming adapters

* Illumina adapters are short nucleotide sequences (oligos) that are used in the construction of the sequencing library
* If the DNA fragment is shorter than the read, then the sequence may "read-through" to the adapter on the other end.
* Adapter contamination: many genomes in Genbank are contaminated with adapter sequences that were not removed by the authors.
* *e.g.*, the [carp genome](http://www.opiniomics.org/we-need-to-stop-making-this-simple-fcking-mistake/)

---

# Software for trimming

* [Trimmomatic](http://www.usadellab.org/cms/?page=trimmomatic) - Java program for trimming Illumina data
* [cutadapt](https://github.com/marcelm/cutadapt) - A Python module for removing adapter sequences and other artefacts
* [AfterQC](https://github.com/OpenGene/AfterQC) - Another Python module for trimming, discarding low quality bases and reconciling paired-end reads.

---

# Sources of error

* PCR error
* Homopolymer errors (454, Ion Torrent)
* Cross-talk between clusters
* Sample cross-contamination
* Inter-run contamination

---

# Homopolymer error

![](/img/K103N.png)

---

# &phi;X174 control

* Bacteriophage genomic DNA (&phi;X174) is commonly used as a positive control ("spike-in") for Illumina runs
* Reads mapping to &phi;X714 are sorted by the vendor software and used to measure run-specific error rates
* Sometimes the software fails to remove all &phi;X174 reads from the data! About [10% of genomes](https://environmentalmicrobiome.biomedcentral.com/articles/10.1186/1944-3277-10-18) published by 2015 contaminated.
* These results are stored in one of the binary `InterOp` files.

---

# InterOp files
* About 11 binary (`*.bin`) InterOp files produces for every Illumina run
* Binary means that these files are not plain-text:
  ```
  7??=??&M7??=??`M7??=???M?>7??M???>a??MW??=&??M?>ݸ=M???>???M?	
  ?=D??M?j">??M???=ȷMw??=??M{?=h?N?a?>???N?Z>???N?+?=?N??=?8N?
  U?=??gN??H>[??N?n>?:@>ŴYN	9iB>l??N?u>#??N
  ```
* Store metrics about that run, *e.g.*, tile-/cycle-mean quality scores
* `ErrorMetricsOut.bin` stores the &phi;X174 error rates.

---

# Bad tile-cycle combos

Extracted error rates from the InterOp file of a run where every HIV-1 patient was diagnosed with the same drug resistance mutation (K103N)

![](/img/bad-cycles1.png)

---

# Cross-contamination

* Reads are incorrectly demultiplexed into other samples.
* "Index-hopping" - the barcode from one end recombines with the barcode from another template.
* Roughly 0.1% to 1% of reads assigned to a sample may come from other samples.

![](/img/crosstalk.png)

<small>
Image from LE MacConaill *et al*. 2018 BMC Genomics 19:30
</small>

---

# Carryover contamination

![](/img/carryover1.png)
![](/img/carryover2.png)

---

# Open-source software (OSS)

* Source code is released under a license that grants users the freedom to use, modify and re-distribute it
* Some licenses require developers using the source code to release *their* code under the same license ("viral" GPL).
* The course materials at http://artpoon.github.io/BioID are released under the [Creative Commons Attribution-ShareAlike](https://creativecommons.org/licenses/by-sa/4.0/) license (same as Wikipedia).

---

# Pros and cons of open-source

* OSS is often *free*, which promotes widespread use, *e.g.*, R.
* Transparency to be inspected by the developer community can make OSS more secure, reliable.
* The majority of bioinformatics software developed by researchers is released as OSS.
* OSS can only as good as it is actively maintained - there are many abandoned projects.
* Proprietary software may be more consistently maintained because developers are paid.

---

# Compiling from source

* If a program is only distributed as source code or you need to customize it, you have to compile the code.
* A compiler (`gcc`, `javac`, `gfortran`) converts instructions that can be read by people into those that can be read by your computer.
* Many programs can be compiled by using the combination `./configure`, `make`,  and `sudo make install`.

---

# Package managers

* A public repository stores binaries (executable files compiled from source) that work for many different systems.
* A user can run a package manager that downloads the required binaries from a repository.
* Much easier than compiling from source, but can lead to unexpected problems.
* *e.g.,* NCBI `sra-toolkit` package was broken for a long time.


