# MIMM4750G
## Diversity and rates of evolution
![](https://imgs.xkcd.com/comics/evolving.png)

---

# Measuring diversity

* Genetic distances are useful
* What if there are many sequences (lots of pairwise comparisons!)
* Next-generation sequencing yields thousands to millions of reads.
* What if the sequences are very long?
* Bacterial genomes can be over 14Mbp long.
* Shortest animal genome is about 19.6Mbp (*Pratylenchus coffeae*, parasitic nematode of plants).

---

# Simple diversity measures

* Proportion of polymorphic sites in an alignment.
* Nucleotide diversity - the expected p-distance ($p$) between a random pair of sequences:
  `$$\pi = \sum_{i}\sum_{j} f_i f_j p_{ij}$$`
  where $f_i$ is the frequency of the $i$-th sequence variant.
* Shannon entropy  

---

# Shannon Entropy

* Based on information theory, Shannon entropy is calculated from the frequencies of variants indexed by $i$:

  $$ S = -\sum_{i} p_i \log(p_i)$$

* If most frequencies are near zero, $S$ approaches 0.
* $S$ is greatest when frequencies are equal.
* Often averaged across nucleotide or amino acid sites of an alignment.

---

Application of Shannon entropy to characterize 16S rRNA gene diversity using PacBio (long read) and MiSeq (short read) NGS platforms.
<img src="/img/Clostridiales.svg" width="600px"/>

<small>
Figure from O Franz&eacute;n *et al.* (2015) Microbiome 3:43.
</small>

---

# Limitations of diversity measures

* Many diversity measures assume every sequence is an independent observation.
* We can badly overestimate diversity if most sequences inherited their variation from the same common ancestors.
* 

---

# Using trees

* Building a tree relating sequences enables us to distinguish between *identity by state* and *identity by descent*.
* Identity by descent:  different infections inherited the variant from their common ancestor.
* Identity by state: variant arose independently in different infections.
* This is key information for reconstructing how many times a mutation occurred.

---

# Rates of evolution

* Counting mutations is the key to measuring the rate of evolution
* **Why do rates matter?**
* Sites that evolve faster than others can reveal targets of selection.
* Rate of evolution may determine which variant survives.
* We can use the rate of evolution to extrapolate back in time.

---

Spread of *Escherichia coli* on a "megaplate" with gradients of antibiotic trimethoprim.
![](/img/megaplate5.gif)

<small>
Media from M Baym *et al* (2016) Science 353: 1147.
</small>

---

HIV-1 protease homodimer, colored by rate of evolution (blue fastest).
Protease inhibitor (yellow) bound at active site.
<img src="/img/foley-protease.jpeg" width="500px"/>

<small>
Image from Kuiken *et al.*, Los Alamos National Laboratory HIV Sequence Compendium 2003.
</small>

---

# Measuring rates of evolution

* We assume a constant rate over time, $\mu$.
* 


